<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 RNA Data preprocessing | IOBR (Immuno-Oncology Biological Research)</title>
  <meta name="description" content="The tutorial of IOBR package" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 RNA Data preprocessing | IOBR (Immuno-Oncology Biological Research)" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://iobr.github.io/book/./fig/cover.png" />
  <meta property="og:description" content="The tutorial of IOBR package" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 RNA Data preprocessing | IOBR (Immuno-Oncology Biological Research)" />
  
  <meta name="twitter:description" content="The tutorial of IOBR package" />
  <meta name="twitter:image" content="https://iobr.github.io/book/./fig/cover.png" />

<meta name="author" content="Dongqiang Zeng, Yiran Fang" />


<meta name="date" content="2024-01-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="how-to-use-iobr.html"/>
<link rel="next" href="signature-score-calculation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The tutorial of IOBR</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><strong>Introduction</strong></a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#introduction-1"><i class="fa fa-check"></i><b>0.1</b> üìö Introduction</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>0.2</b> üíø License</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#previous-publication"><i class="fa fa-check"></i><b>0.3</b> üèÜ Previous publication</a></li>
<li class="chapter" data-level="0.4" data-path="index.html"><a href="index.html#major-updates"><i class="fa fa-check"></i><b>0.4</b> ‚è≥ Major Updates</a></li>
<li class="chapter" data-level="0.5" data-path="index.html"><a href="index.html#reporting-bugs"><i class="fa fa-check"></i><b>0.5</b> ‚úâÔ∏è Reporting bugs</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html"><i class="fa fa-check"></i><b>1</b> <strong>How to install IOBR</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#install-dependency-packages"><i class="fa fa-check"></i><b>1.1</b> ‚è≥ Install Dependency Packages</a></li>
<li class="chapter" data-level="1.2" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#install-iobr-package"><i class="fa fa-check"></i><b>1.2</b> üíª Install IOBR package</a></li>
<li class="chapter" data-level="1.3" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#how-to-update-iobr"><i class="fa fa-check"></i><b>1.3</b> üìç How to update IOBR</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="how-to-use-iobr.html"><a href="how-to-use-iobr.html"><i class="fa fa-check"></i><b>2</b> <strong>How to use IOBR</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-to-use-iobr.html"><a href="how-to-use-iobr.html#the-main-pipeline-of-iobr"><i class="fa fa-check"></i><b>2.1</b> üîç The main pipeline of IOBR</a></li>
<li class="chapter" data-level="2.2" data-path="how-to-use-iobr.html"><a href="how-to-use-iobr.html#main-functions-of-iobr"><i class="fa fa-check"></i><b>2.2</b> üóÇ Main Functions of IOBR</a></li>
<li class="chapter" data-level="2.3" data-path="how-to-use-iobr.html"><a href="how-to-use-iobr.html#current-working-environment"><i class="fa fa-check"></i><b>2.3</b> üåê Current working environment</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html"><i class="fa fa-check"></i><b>3</b> <strong>RNA Data preprocessing</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#loading-packages"><i class="fa fa-check"></i><b>3.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.2" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#download-array-data-using-geoquery"><i class="fa fa-check"></i><b>3.2</b> Download array data using <code>GEOquery</code></a></li>
<li class="chapter" data-level="3.3" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#gene-annotation"><i class="fa fa-check"></i><b>3.3</b> Gene Annotation</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#for-array-data-hgu133plus-2-affaymetrix"><i class="fa fa-check"></i><b>3.3.1</b> For Array data: HGU133PLUS-2 (Affaymetrix)</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#download-rnaseq-data-using-ucscxenatools"><i class="fa fa-check"></i><b>3.4</b> Download RNAseq data using <code>UCSCXenaTools</code></a></li>
<li class="chapter" data-level="3.5" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#normalization-and-gene-annotation"><i class="fa fa-check"></i><b>3.5</b> Normalization and Gene annotation</a></li>
<li class="chapter" data-level="3.6" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#identifying-outlier-samples"><i class="fa fa-check"></i><b>3.6</b> Identifying outlier samples</a></li>
<li class="chapter" data-level="3.7" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#pca-analysis-of-molecular-subtypes"><i class="fa fa-check"></i><b>3.7</b> PCA analysis of molecular subtypes</a></li>
<li class="chapter" data-level="3.8" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#batch-effect-correction"><i class="fa fa-check"></i><b>3.8</b> Batch effect correction</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#for-microarray-data"><i class="fa fa-check"></i><b>3.8.1</b> For microarray data</a></li>
<li class="chapter" data-level="3.8.2" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#for-rnaseq-count-data"><i class="fa fa-check"></i><b>3.8.2</b> For RNAseq count data</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#references"><i class="fa fa-check"></i><b>3.9</b> References</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html"><i class="fa fa-check"></i><b>4</b> <strong>Signature Score Calculation</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#loading-packages-1"><i class="fa fa-check"></i><b>4.1</b> Loading packages</a></li>
<li class="chapter" data-level="4.2" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#downloading-data-for-example"><i class="fa fa-check"></i><b>4.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="4.3" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#signature-score-estimation"><i class="fa fa-check"></i><b>4.3</b> Signature score estimation</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#signature-collection-of-iobr"><i class="fa fa-check"></i><b>4.3.1</b> Signature collection of IOBR</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#estimation-of-signature-using-pca-method"><i class="fa fa-check"></i><b>4.4</b> Estimation of signature using PCA method</a></li>
<li class="chapter" data-level="4.5" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#estimated-using-the-ssgsea-methodology"><i class="fa fa-check"></i><b>4.5</b> Estimated using the ssGSEA methodology</a></li>
<li class="chapter" data-level="4.6" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#calculated-using-the-z-score-function."><i class="fa fa-check"></i><b>4.6</b> Calculated using the z-score function.</a></li>
<li class="chapter" data-level="4.7" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#calculated-using-all-three-methods-at-the-same-time"><i class="fa fa-check"></i><b>4.7</b> Calculated using all three methods at the same time</a></li>
<li class="chapter" data-level="4.8" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#classification-of-signatures"><i class="fa fa-check"></i><b>4.8</b> Classification of signatures</a></li>
<li class="chapter" data-level="4.9" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#how-to-customise-the-signature-gene-list-for-calculate_signature_score"><i class="fa fa-check"></i><b>4.9</b> How to customise the signature gene list for <code>calculate_signature_score</code></a>
<ul>
<li class="chapter" data-level="4.9.1" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#method-1-use-excel-for-storage-and-construction"><i class="fa fa-check"></i><b>4.9.1</b> Method-1: Use excel for storage and construction</a></li>
<li class="chapter" data-level="4.9.2" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#method-2-build-the-list-structure-directly"><i class="fa fa-check"></i><b>4.9.2</b> Method-2: Build the list structure directly</a></li>
<li class="chapter" data-level="4.9.3" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#method3-add-the-new-signature-to-the-existing-gene-list"><i class="fa fa-check"></i><b>4.9.3</b> Method3: Add the new signature to the existing gene list</a></li>
</ul></li>
<li class="chapter" data-level="4.10" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#how-to-export-gene-signature"><i class="fa fa-check"></i><b>4.10</b> How to export gene signature</a></li>
<li class="chapter" data-level="4.11" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#references-1"><i class="fa fa-check"></i><b>4.11</b> References</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html"><i class="fa fa-check"></i><b>5</b> <strong>TME deconvolution</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#loading-packages-2"><i class="fa fa-check"></i><b>5.1</b> Loading packages</a></li>
<li class="chapter" data-level="5.2" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#downloading-data-for-example-1"><i class="fa fa-check"></i><b>5.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="5.3" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#available-methods-to-decode-tme-contexture"><i class="fa fa-check"></i><b>5.3</b> Available Methods to Decode TME Contexture</a></li>
<li class="chapter" data-level="5.4" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-1-cibersort"><i class="fa fa-check"></i><b>5.4</b> Method 1: CIBERSORT</a></li>
<li class="chapter" data-level="5.5" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-2-epic"><i class="fa fa-check"></i><b>5.5</b> Method 2: EPIC</a></li>
<li class="chapter" data-level="5.6" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-3-mcpcounter"><i class="fa fa-check"></i><b>5.6</b> Method 3: MCPcounter</a></li>
<li class="chapter" data-level="5.7" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-4-xcell"><i class="fa fa-check"></i><b>5.7</b> Method 4: xCELL</a></li>
<li class="chapter" data-level="5.8" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-5-estimate"><i class="fa fa-check"></i><b>5.8</b> Method 5: ESTIMATE</a></li>
<li class="chapter" data-level="5.9" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-6-timer"><i class="fa fa-check"></i><b>5.9</b> Method 6: TIMER</a></li>
<li class="chapter" data-level="5.10" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-7-quantiseq"><i class="fa fa-check"></i><b>5.10</b> Method 7: quanTIseq</a></li>
<li class="chapter" data-level="5.11" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-8-ips"><i class="fa fa-check"></i><b>5.11</b> Method 8: IPS</a></li>
<li class="chapter" data-level="5.12" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#combination-of-above-deconvolution-results"><i class="fa fa-check"></i><b>5.12</b> Combination of above deconvolution results</a></a></li>
<li class="chapter" data-level="5.13" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#how-to-customise-the-signature-matrix-for-svr-and-lesi-algorithm"><i class="fa fa-check"></i><b>5.13</b> How to customise the signature matrix for <code>SVR</code> and <code>lesi</code> algorithm</a></li>
<li class="chapter" data-level="5.14" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#references-2"><i class="fa fa-check"></i><b>5.14</b> References</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html"><i class="fa fa-check"></i><b>6</b> <strong>Signature Score and Relevant phenotypes</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#loading-packages-3"><i class="fa fa-check"></i><b>6.1</b> Loading packages</a></li>
<li class="chapter" data-level="6.2" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#downloading-data-for-example-2"><i class="fa fa-check"></i><b>6.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="6.3" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#gene-annotation-1"><i class="fa fa-check"></i><b>6.3</b> Gene Annotation</a></li>
<li class="chapter" data-level="6.4" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#estimation-of-signatures"><i class="fa fa-check"></i><b>6.4</b> Estimation of signatures</a></li>
<li class="chapter" data-level="6.5" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#combining-score-data-and-phenotype-data"><i class="fa fa-check"></i><b>6.5</b> Combining score data and phenotype data</a></li>
<li class="chapter" data-level="6.6" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#identifying-features-associated-with-survival"><i class="fa fa-check"></i><b>6.6</b> Identifying features associated with survival</a></li>
<li class="chapter" data-level="6.7" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#visulization-using-heatmap"><i class="fa fa-check"></i><b>6.7</b> Visulization using heatmap</a></li>
<li class="chapter" data-level="6.8" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#focus-on-target-signatures"><i class="fa fa-check"></i><b>6.8</b> Focus on target signatures</a></li>
<li class="chapter" data-level="6.9" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#survival-analysis-and-visulization"><i class="fa fa-check"></i><b>6.9</b> Survival analysis and visulization</a>
<ul>
<li class="chapter" data-level="6.9.1" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#kaplan-meier-plot"><i class="fa fa-check"></i><b>6.9.1</b> Kaplan-Meier plot</a></li>
<li class="chapter" data-level="6.9.2" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#time-dependent-roc-curve"><i class="fa fa-check"></i><b>6.9.2</b> Time-Dependent ROC curve</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#batch-correlation-analysis"><i class="fa fa-check"></i><b>6.10</b> Batch correlation analysis</a>
<ul>
<li class="chapter" data-level="6.10.1" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#finding-continuity-variables-associated-with-signatures"><i class="fa fa-check"></i><b>6.10.1</b> Finding continuity variables associated with signatures</a></li>
<li class="chapter" data-level="6.10.2" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#identifying-category-variables-linked-to-signatures"><i class="fa fa-check"></i><b>6.10.2</b> Identifying Category Variables Linked to Signatures</a></li>
<li class="chapter" data-level="6.10.3" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#for-multicategorical-variables-2-subgroups"><i class="fa fa-check"></i><b>6.10.3</b> For multicategorical variables (&gt;2 subgroups)</a></li>
</ul></li>
<li class="chapter" data-level="6.11" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#reference"><i class="fa fa-check"></i><b>6.11</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html"><i class="fa fa-check"></i><b>7</b> <strong>TME Interaction analysis</strong></a>
<ul>
<li class="chapter" data-level="7.1" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#loading-packages-4"><i class="fa fa-check"></i><b>7.1</b> Loading packages</a></li>
<li class="chapter" data-level="7.2" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#downloading-data-for-example-3"><i class="fa fa-check"></i><b>7.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="7.3" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#gene-annotation-hgu133plus-2-affaymetrix"><i class="fa fa-check"></i><b>7.3</b> Gene Annotation: HGU133PLUS-2 (Affaymetrix)</a></li>
<li class="chapter" data-level="7.4" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#tme-deconvolution-using-cibersort-algorithm"><i class="fa fa-check"></i><b>7.4</b> TME deconvolution using CIBERSORT algorithm</a></li>
<li class="chapter" data-level="7.5" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#identifying-tme-patterns"><i class="fa fa-check"></i><b>7.5</b> Identifying TME patterns</a></li>
<li class="chapter" data-level="7.6" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#cell-abundance-of-each-cluster"><i class="fa fa-check"></i><b>7.6</b> Cell abundance of each cluster</a></li>
<li class="chapter" data-level="7.7" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#deg-analysis-between-tme-subtypes"><i class="fa fa-check"></i><b>7.7</b> DEG analysis between TME subtypes</a></li>
<li class="chapter" data-level="7.8" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#identifying-signatures-associated-with-tme-clusters"><i class="fa fa-check"></i><b>7.8</b> Identifying signatures associated with TME clusters</a></li>
<li class="chapter" data-level="7.9" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#references-3"><i class="fa fa-check"></i><b>7.9</b> References</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html"><i class="fa fa-check"></i><b>8</b> <strong>Tumor ecosystem analysis</strong></a>
<ul>
<li class="chapter" data-level="8.1" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#loading-packages-5"><i class="fa fa-check"></i><b>8.1</b> Loading packages</a></li>
<li class="chapter" data-level="8.2" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#downloading-data-for-example-4"><i class="fa fa-check"></i><b>8.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="8.3" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#gene-annotation-hgu133plus-2-affaymetrix-1"><i class="fa fa-check"></i><b>8.3</b> Gene Annotation: HGU133PLUS-2 (Affaymetrix)</a></li>
<li class="chapter" data-level="8.4" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#determine-tme-subtype-of-gastric-cancer-using-tmeclassifier-r-package"><i class="fa fa-check"></i><b>8.4</b> Determine TME subtype of gastric cancer using TMEclassifier R package</a></li>
<li class="chapter" data-level="8.5" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#deg-analysis-method1"><i class="fa fa-check"></i><b>8.5</b> DEG analysis: method1</a></li>
<li class="chapter" data-level="8.6" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#gsea-analysis-based-on-differential-express-gene-analysis-results"><i class="fa fa-check"></i><b>8.6</b> GSEA analysis based on differential express gene analysis results</a></li>
<li class="chapter" data-level="8.7" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#deg-analysis-method2"><i class="fa fa-check"></i><b>8.7</b> DEG analysis: method2</a></li>
<li class="chapter" data-level="8.8" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#identifying-signatures-associated-with-tme-clusters-1"><i class="fa fa-check"></i><b>8.8</b> Identifying signatures associated with TME clusters</a></li>
<li class="chapter" data-level="8.9" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#references-4"><i class="fa fa-check"></i><b>8.9</b> References</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html"><i class="fa fa-check"></i><b>9</b> <strong>TME and genomic interaction</strong></a>
<ul>
<li class="chapter" data-level="9.1" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#loading-packages-6"><i class="fa fa-check"></i><b>9.1</b> Loading packages</a></li>
<li class="chapter" data-level="9.2" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#genomic-data-prepare"><i class="fa fa-check"></i><b>9.2</b> Genomic data prepare</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#using-tcgabiolinks-to-download-genomic-data"><i class="fa fa-check"></i><b>9.2.1</b> Using TCGAbiolinks to download genomic data</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#identifying-mutations-associated-with-tme"><i class="fa fa-check"></i><b>9.3</b> Identifying Mutations Associated with TME</a></li>
<li class="chapter" data-level="9.4" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#oncoprint-of-result"><i class="fa fa-check"></i><b>9.4</b> OncoPrint of result</a></li>
<li class="chapter" data-level="9.5" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#boxplot-of-top-10-mutated-genes"><i class="fa fa-check"></i><b>9.5</b> Boxplot of top 10 mutated genes</a></li>
<li class="chapter" data-level="9.6" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#other-methods-to-obtaind-genomic-data"><i class="fa fa-check"></i><b>9.6</b> Other methods to obtaind genomic data</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#using-tcgamutations"><i class="fa fa-check"></i><b>9.6.1</b> Using TCGAmutations</a></li>
<li class="chapter" data-level="9.6.2" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#using-maftools"><i class="fa fa-check"></i><b>9.6.2</b> Using maftools</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#references-5"><i class="fa fa-check"></i><b>9.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="tme-modeling.html"><a href="tme-modeling.html"><i class="fa fa-check"></i><b>10</b> <strong>TME Modeling</strong></a>
<ul>
<li class="chapter" data-level="10.1" data-path="tme-modeling.html"><a href="tme-modeling.html#loading-packages-7"><i class="fa fa-check"></i><b>10.1</b> Loading packages</a></li>
<li class="chapter" data-level="10.2" data-path="tme-modeling.html"><a href="tme-modeling.html#data-prepare"><i class="fa fa-check"></i><b>10.2</b> Data prepare</a></li>
<li class="chapter" data-level="10.3" data-path="tme-modeling.html"><a href="tme-modeling.html#input-data-overall-survival-prepare"><i class="fa fa-check"></i><b>10.3</b> Input data (overall survival) prepare</a></li>
<li class="chapter" data-level="10.4" data-path="tme-modeling.html"><a href="tme-modeling.html#constructing-survival-prediction-models"><i class="fa fa-check"></i><b>10.4</b> Constructing survival prediction models</a></li>
<li class="chapter" data-level="10.5" data-path="tme-modeling.html"><a href="tme-modeling.html#input-data-response-prepare"><i class="fa fa-check"></i><b>10.5</b> Input data (Response) prepare</a></li>
<li class="chapter" data-level="10.6" data-path="tme-modeling.html"><a href="tme-modeling.html#constructing-prediction-models-for-response"><i class="fa fa-check"></i><b>10.6</b> Constructing prediction models for response</a></li>
<li class="chapter" data-level="10.7" data-path="tme-modeling.html"><a href="tme-modeling.html#references-6"><i class="fa fa-check"></i><b>10.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html"><i class="fa fa-check"></i><b>11</b> <strong>Some tips about IOBR</strong></a>
<ul>
<li class="chapter" data-level="11.1" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html#loading-packages-8"><i class="fa fa-check"></i><b>11.1</b> Loading packages</a></li>
<li class="chapter" data-level="11.2" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html#colour-configuration"><i class="fa fa-check"></i><b>11.2</b> Colour Configuration</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html#categorical-variable"><i class="fa fa-check"></i><b>11.2.1</b> categorical variable</a></li>
<li class="chapter" data-level="11.2.2" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html#higher-number-of-subgroups"><i class="fa fa-check"></i><b>11.2.2</b> Higher number of subgroups</a></li>
<li class="chapter" data-level="11.2.3" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html#gradient-colour-or-heatmap-colour-scheme"><i class="fa fa-check"></i><b>11.2.3</b> Gradient colour or heatmap colour scheme</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="references-7.html"><a href="references-7.html"><i class="fa fa-check"></i><b>12</b> <strong>References</strong></a>
<ul>
<li class="chapter" data-level="12.1" data-path="references-7.html"><a href="references-7.html#tme-deconvolution-1"><i class="fa fa-check"></i><b>12.1</b> TME deconvolution</a></li>
<li class="chapter" data-level="12.2" data-path="references-7.html"><a href="references-7.html#tme-signatures"><i class="fa fa-check"></i><b>12.2</b> TME Signatures</a></li>
<li class="chapter" data-level="12.3" data-path="references-7.html"><a href="references-7.html#data-sets"><i class="fa fa-check"></i><b>12.3</b> Data sets</a></li>
<li class="chapter" data-level="12.4" data-path="references-7.html"><a href="references-7.html#others"><i class="fa fa-check"></i><b>12.4</b> Others</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/DongqiangZeng0808" target="blank">Welcome to my GitHub</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">IOBR (Immuno-Oncology Biological Research)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="rna-data-preprocessing" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> <strong>RNA Data preprocessing</strong><a href="rna-data-preprocessing.html#rna-data-preprocessing" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="loading-packages" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Loading packages<a href="rna-data-preprocessing.html#loading-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Load the IOBR package in your R session after the installation is complete:</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="rna-data-preprocessing.html#cb39-1" tabindex="-1"></a><span class="fu">library</span>(IOBR)</span>
<span id="cb39-2"><a href="rna-data-preprocessing.html#cb39-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb39-3"><a href="rna-data-preprocessing.html#cb39-3" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span></code></pre></div>
</div>
<div id="download-array-data-using-geoquery" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Download array data using <code>GEOquery</code><a href="rna-data-preprocessing.html#download-array-data-using-geoquery" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Obtaining data set from GEO <a href="https://pubmed.ncbi.nlm.nih.gov/25894828/">Gastric cancer: GSE62254</a> using <code>GEOquery</code> R package.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="rna-data-preprocessing.html#cb40-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;GEOquery&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;GEOquery&quot;</span>)</span>
<span id="cb40-2"><a href="rna-data-preprocessing.html#cb40-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;GEOquery&quot;</span>)</span>
<span id="cb40-3"><a href="rna-data-preprocessing.html#cb40-3" tabindex="-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: This process may take a few minutes which depends on the internet connection speed. Please wait for its completion.</span></span>
<span id="cb40-4"><a href="rna-data-preprocessing.html#cb40-4" tabindex="-1"></a>eset_geo<span class="ot">&lt;-</span><span class="fu">getGEO</span>(<span class="at">GEO     =</span> <span class="st">&quot;GSE62254&quot;</span>, <span class="at">getGPL  =</span> F, <span class="at">destdir =</span> <span class="st">&quot;./&quot;</span>)</span>
<span id="cb40-5"><a href="rna-data-preprocessing.html#cb40-5" tabindex="-1"></a>eset    <span class="ot">&lt;-</span>eset_geo[[<span class="dv">1</span>]]</span>
<span id="cb40-6"><a href="rna-data-preprocessing.html#cb40-6" tabindex="-1"></a>eset    <span class="ot">&lt;-</span><span class="fu">exprs</span>(eset)</span>
<span id="cb40-7"><a href="rna-data-preprocessing.html#cb40-7" tabindex="-1"></a>eset[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##           GSM1523727 GSM1523728 GSM1523729 GSM1523744 GSM1523745
## 1007_s_at  3.2176645  3.0624323  3.0279131   2.921683  2.8456013
## 1053_at    2.4050109  2.4394879  2.2442708   2.345916  2.4328582
## 117_at     1.4933412  1.8067380  1.5959665   1.839822  1.8326058
## 121_at     2.1965561  2.2812181  2.1865556   2.258599  2.1874363
## 1255_g_at  0.8698382  0.9502466  0.8125414   1.012860  0.9441993</code></pre>
</div>
<div id="gene-annotation" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Gene Annotation<a href="rna-data-preprocessing.html#gene-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Annotation of genes in the expression matrix and removal of duplicate genes.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="rna-data-preprocessing.html#cb42-1" tabindex="-1"></a><span class="co"># Load the annotation file `anno_hug133plus2` in IOBR.</span></span>
<span id="cb42-2"><a href="rna-data-preprocessing.html#cb42-2" tabindex="-1"></a><span class="fu">head</span>(anno_hug133plus2)</span></code></pre></div>
<pre><code>## # A tibble: 6 √ó 2
##   probe_id  symbol 
##   &lt;fct&gt;     &lt;fct&gt;  
## 1 1007_s_at MIR4640
## 2 1053_at   RFC2   
## 3 117_at    HSPA6  
## 4 121_at    PAX8   
## 5 1255_g_at GUCA1A 
## 6 1294_at   MIR5193</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="rna-data-preprocessing.html#cb44-1" tabindex="-1"></a><span class="co"># Load the annotation file `anno_grch38` in IOBR.</span></span>
<span id="cb44-2"><a href="rna-data-preprocessing.html#cb44-2" tabindex="-1"></a><span class="fu">head</span>(anno_grch38)</span></code></pre></div>
<pre><code>##                id eff_length        gc entrez   symbol chr     start       end
## 1 ENSG00000000003       4536 0.3992504   7105   TSPAN6   X 100627109 100639991
## 2 ENSG00000000005       1476 0.4241192  64102     TNMD   X 100584802 100599885
## 3 ENSG00000000419       9276 0.4252911   8813     DPM1  20  50934867  50958555
## 4 ENSG00000000457       6883 0.4117391  57147    SCYL3   1 169849631 169894267
## 5 ENSG00000000460       5970 0.4298157  55732 C1orf112   1 169662007 169854080
## 6 ENSG00000000938       3382 0.5644589   2268      FGR   1  27612064  27635277
##   strand        biotype
## 1     -1 protein_coding
## 2      1 protein_coding
## 3     -1 protein_coding
## 4     -1 protein_coding
## 5      1 protein_coding
## 6     -1 protein_coding
##                                                                                                  description
## 1                                                          tetraspanin 6 [Source:HGNC Symbol;Acc:HGNC:11858]
## 2                                                            tenomodulin [Source:HGNC Symbol;Acc:HGNC:17757]
## 3 dolichyl-phosphate mannosyltransferase polypeptide 1, catalytic subunit [Source:HGNC Symbol;Acc:HGNC:3005]
## 4                                               SCY1-like, kinase-like 3 [Source:HGNC Symbol;Acc:HGNC:19285]
## 5                                    chromosome 1 open reading frame 112 [Source:HGNC Symbol;Acc:HGNC:25565]
## 6                          FGR proto-oncogene, Src family tyrosine kinase [Source:HGNC Symbol;Acc:HGNC:3697]</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="rna-data-preprocessing.html#cb46-1" tabindex="-1"></a><span class="co"># Load the annotation file `anno_gc_vm32` in IOBR for mouse RNAseq data</span></span>
<span id="cb46-2"><a href="rna-data-preprocessing.html#cb46-2" tabindex="-1"></a><span class="fu">head</span>(anno_gc_vm32)</span></code></pre></div>
<pre><code>##                   id eff_length        gc symbol      mgi_id      gene_type
## 1 ENSMUSG00000000001       3262 0.4350092  Gnai3   MGI:95773 protein_coding
## 2 ENSMUSG00000000003        902 0.3481153   Pbsn MGI:1860484 protein_coding
## 3 ENSMUSG00000000028       3506 0.4962921  Cdc45 MGI:1338073 protein_coding
## 4 ENSMUSG00000000031       2625 0.5588571    H19   MGI:95891         lncRNA
## 5 ENSMUSG00000000037       6397 0.4377052  Scml2 MGI:1340042 protein_coding
## 6 ENSMUSG00000000049       1594 0.5050188   Apoh   MGI:88058 protein_coding
##       start       end transcript_id  ont
## 1 108014596 108053462          &lt;NA&gt; &lt;NA&gt;
## 2  76881507  76897229          &lt;NA&gt; &lt;NA&gt;
## 3  18599197  18630737          &lt;NA&gt; &lt;NA&gt;
## 4 142129262 142131886          &lt;NA&gt; &lt;NA&gt;
## 5 159865521 160041209          &lt;NA&gt; &lt;NA&gt;
## 6 108234180 108305222          &lt;NA&gt; &lt;NA&gt;</code></pre>
<div id="for-array-data-hgu133plus-2-affaymetrix" class="section level3 hasAnchor" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> For Array data: HGU133PLUS-2 (Affaymetrix)<a href="rna-data-preprocessing.html#for-array-data-hgu133plus-2-affaymetrix" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="rna-data-preprocessing.html#cb48-1" tabindex="-1"></a><span class="co"># Conduct gene annotation using `anno_hug133plus2` file; If identical gene symbols exists, these genes would be ordered by the mean expression levels. The gene symbol with highest mean expression level is selected and remove others. </span></span>
<span id="cb48-2"><a href="rna-data-preprocessing.html#cb48-2" tabindex="-1"></a></span>
<span id="cb48-3"><a href="rna-data-preprocessing.html#cb48-3" tabindex="-1"></a>eset<span class="ot">&lt;-</span><span class="fu">anno_eset</span>(<span class="at">eset       =</span> eset,</span>
<span id="cb48-4"><a href="rna-data-preprocessing.html#cb48-4" tabindex="-1"></a>                <span class="at">annotation =</span> anno_hug133plus2,</span>
<span id="cb48-5"><a href="rna-data-preprocessing.html#cb48-5" tabindex="-1"></a>                <span class="at">symbol     =</span> <span class="st">&quot;symbol&quot;</span>,</span>
<span id="cb48-6"><a href="rna-data-preprocessing.html#cb48-6" tabindex="-1"></a>                <span class="at">probe      =</span> <span class="st">&quot;probe_id&quot;</span>,</span>
<span id="cb48-7"><a href="rna-data-preprocessing.html#cb48-7" tabindex="-1"></a>                <span class="at">method     =</span> <span class="st">&quot;mean&quot;</span>)</span>
<span id="cb48-8"><a href="rna-data-preprocessing.html#cb48-8" tabindex="-1"></a>eset[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>##              GSM1523727 GSM1523728 GSM1523729
## SH3KBP1        4.327974   4.316195   4.351425
## RPL41          4.246149   4.246808   4.257940
## EEF1A1         4.293762   4.291038   4.262199
## COX2           4.250288   4.283714   4.270508
## LOC101928826   4.219303   4.219670   4.213252</code></pre>
</div>
</div>
<div id="download-rnaseq-data-using-ucscxenatools" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Download RNAseq data using <code>UCSCXenaTools</code><a href="rna-data-preprocessing.html#download-rnaseq-data-using-ucscxenatools" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this section, we are going to download RNA-seq data from The Cancer Genome Atlas (TCGA) for applying the downstream analysis workflow of <strong>IOBR</strong>. Particularly, we will use the convenient R package <a href="https://cran.r-project.org/web/packages/UCSCXenaTools/vignettes/USCSXenaTools.html"><strong>UCSCXenaTools</strong></a> to query and download the RNA-seq data of TCGA stomach cancer cohort.</p>
<p>Use the following code to check and install <strong>UCSCXenaTools</strong>.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="rna-data-preprocessing.html#cb50-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;UCSCXenaTools&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))  </span>
<span id="cb50-2"><a href="rna-data-preprocessing.html#cb50-2" tabindex="-1"></a>  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;UCSCXenaTools&quot;</span>)</span></code></pre></div>
<p><strong>UCSCXenaTools</strong> provides an R interface to access public cancer datasets from <a href="https://xenabrowser.net/datapages/">UCSC Xena data hubs</a>, including multiple pan-cancer studies like TCGA and PCAWG. You can directly access information of all datasets in R.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="rna-data-preprocessing.html#cb51-1" tabindex="-1"></a><span class="fu">library</span>(UCSCXenaTools)</span></code></pre></div>
<pre><code>## Warning: package &#39;UCSCXenaTools&#39; was built under R version 4.2.1</code></pre>
<pre><code>## =========================================================================================
## UCSCXenaTools version 1.4.8
## Project URL: https://github.com/ropensci/UCSCXenaTools
## Usages: https://cran.r-project.org/web/packages/UCSCXenaTools/vignettes/USCSXenaTools.html
## 
## If you use it in published research, please cite:
## Wang et al., (2019). The UCSCXenaTools R package: a toolkit for accessing genomics data
##   from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq.
##   Journal of Open Source Software, 4(40), 1627, https://doi.org/10.21105/joss.01627
## =========================================================================================
##                               --Enjoy it--</code></pre>
<pre><code>## 
## Attaching package: &#39;UCSCXenaTools&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:Biobase&#39;:
## 
##     samples</code></pre>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="rna-data-preprocessing.html#cb56-1" tabindex="-1"></a><span class="fu">head</span>(XenaData)</span></code></pre></div>
<pre><code>## # A tibble: 6 √ó 17
##   XenaHosts XenaHostNames XenaCohorts XenaDatasets SampleCount DataSubtype Label
##   &lt;chr&gt;     &lt;chr&gt;         &lt;chr&gt;       &lt;chr&gt;              &lt;int&gt; &lt;chr&gt;       &lt;chr&gt;
## 1 https://‚Ä¶ publicHub     Breast Can‚Ä¶ ucsfNeve_pu‚Ä¶          51 gene expre‚Ä¶ Neve‚Ä¶
## 2 https://‚Ä¶ publicHub     Breast Can‚Ä¶ ucsfNeve_pu‚Ä¶          57 phenotype   Phen‚Ä¶
## 3 https://‚Ä¶ publicHub     Glioma (Ko‚Ä¶ kotliarov20‚Ä¶         194 copy number Kotl‚Ä¶
## 4 https://‚Ä¶ publicHub     Glioma (Ko‚Ä¶ kotliarov20‚Ä¶         194 phenotype   Phen‚Ä¶
## 5 https://‚Ä¶ publicHub     Lung Cance‚Ä¶ weir2007_pu‚Ä¶         383 copy number CGH  
## 6 https://‚Ä¶ publicHub     Lung Cance‚Ä¶ weir2007_pu‚Ä¶         383 phenotype   Phen‚Ä¶
## # ‚Ñπ 10 more variables: Type &lt;chr&gt;, AnatomicalOrigin &lt;chr&gt;, SampleType &lt;chr&gt;,
## #   Tags &lt;chr&gt;, ProbeMap &lt;chr&gt;, LongTitle &lt;chr&gt;, Citation &lt;chr&gt;, Version &lt;chr&gt;,
## #   Unit &lt;chr&gt;, Platform &lt;chr&gt;</code></pre>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="rna-data-preprocessing.html#cb58-1" tabindex="-1"></a><span class="co"># You can use view(XenaData) to find your dataset of interest</span></span></code></pre></div>
<p><strong>UCSCXenaTools</strong> provides <a href="https://cran.r-project.org/web/packages/UCSCXenaTools/vignettes/USCSXenaTools.html#workflow">workflow functions</a> to generate object, filter, query, download and load the dataset(s) of interest. The following code show a standardized <strong>UCSCXenaTools</strong> data workflow to query the data from UCSC Xena data hub and load it into R.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="rna-data-preprocessing.html#cb59-1" tabindex="-1"></a><span class="fu">library</span>(UCSCXenaTools)</span>
<span id="cb59-2"><a href="rna-data-preprocessing.html#cb59-2" tabindex="-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: This process may take a few minutes which depends on the internet connection speed. Please wait for its completion.</span></span>
<span id="cb59-3"><a href="rna-data-preprocessing.html#cb59-3" tabindex="-1"></a>eset_stad<span class="ot">&lt;-</span><span class="fu">XenaGenerate</span>(<span class="at">subset =</span> XenaCohorts <span class="sc">==</span><span class="st">&quot;GDC TCGA Stomach Cancer (STAD)&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb59-4"><a href="rna-data-preprocessing.html#cb59-4" tabindex="-1"></a>  <span class="fu">XenaFilter</span>(<span class="at">filterDatasets    =</span> <span class="st">&quot;TCGA-STAD.htseq_counts.tsv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb59-5"><a href="rna-data-preprocessing.html#cb59-5" tabindex="-1"></a>  <span class="fu">XenaQuery</span>() <span class="sc">%&gt;%</span></span>
<span id="cb59-6"><a href="rna-data-preprocessing.html#cb59-6" tabindex="-1"></a>  <span class="fu">XenaDownload</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb59-7"><a href="rna-data-preprocessing.html#cb59-7" tabindex="-1"></a>  <span class="fu">XenaPrepare</span>()</span>
<span id="cb59-8"><a href="rna-data-preprocessing.html#cb59-8" tabindex="-1"></a>eset_stad[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<p>As the metadata of this dataset have been stored in the <code>XeneData</code> data.frame. You can easily recheck the dataset with code.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="rna-data-preprocessing.html#cb60-1" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">filter</span>(XenaData, XenaDatasets <span class="sc">==</span> <span class="st">&quot;TCGA-STAD.htseq_counts.tsv&quot;</span>) <span class="sc">|&gt;</span> </span>
<span id="cb60-2"><a href="rna-data-preprocessing.html#cb60-2" tabindex="-1"></a>  <span class="fu">as.list</span>()</span></code></pre></div>
</div>
<div id="normalization-and-gene-annotation" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> Normalization and Gene annotation<a href="rna-data-preprocessing.html#normalization-and-gene-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Transform gene expression matrix into TPM format, and conduct subsequent annotation.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="rna-data-preprocessing.html#cb61-1" tabindex="-1"></a><span class="co"># Remove the version numbers in Ensembl ID.</span></span>
<span id="cb61-2"><a href="rna-data-preprocessing.html#cb61-2" tabindex="-1"></a>eset_stad<span class="sc">$</span>Ensembl_ID<span class="ot">&lt;-</span><span class="fu">substring</span>(eset_stad<span class="sc">$</span>Ensembl_ID, <span class="dv">1</span>, <span class="dv">15</span>)</span>
<span id="cb61-3"><a href="rna-data-preprocessing.html#cb61-3" tabindex="-1"></a>eset_stad<span class="ot">&lt;-</span><span class="fu">column_to_rownames</span>(eset_stad, <span class="at">var =</span> <span class="st">&quot;Ensembl_ID&quot;</span>)</span>
<span id="cb61-4"><a href="rna-data-preprocessing.html#cb61-4" tabindex="-1"></a></span>
<span id="cb61-5"><a href="rna-data-preprocessing.html#cb61-5" tabindex="-1"></a><span class="co"># Revert back to original format because the data from UCSC was log2(x+1)transformed.</span></span>
<span id="cb61-6"><a href="rna-data-preprocessing.html#cb61-6" tabindex="-1"></a>eset_stad<span class="ot">&lt;-</span>(<span class="dv">2</span><span class="sc">^</span>eset_stad)<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb61-7"><a href="rna-data-preprocessing.html#cb61-7" tabindex="-1"></a></span>
<span id="cb61-8"><a href="rna-data-preprocessing.html#cb61-8" tabindex="-1"></a>eset_stad<span class="ot">&lt;-</span><span class="fu">count2tpm</span>(<span class="at">countMat =</span> eset_stad, <span class="at">idType =</span> <span class="st">&quot;Ensembl&quot;</span>, <span class="at">org=</span><span class="st">&quot;hsa&quot;</span>, <span class="at">source =</span> <span class="st">&quot;local&quot;</span> )</span>
<span id="cb61-9"><a href="rna-data-preprocessing.html#cb61-9" tabindex="-1"></a></span>
<span id="cb61-10"><a href="rna-data-preprocessing.html#cb61-10" tabindex="-1"></a>eset_stad[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
</div>
<div id="identifying-outlier-samples" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> Identifying outlier samples<a href="rna-data-preprocessing.html#identifying-outlier-samples" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Take ACRG microarray data for example</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="rna-data-preprocessing.html#cb62-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">find_outlier_samples</span>(<span class="at">eset =</span> eset, <span class="at">project =</span> <span class="st">&quot;ACRG&quot;</span>, <span class="at">show_plot =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="data-preprocessing_files/figure-html/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>## [1] &quot;GSM1523817&quot; &quot;GSM1523858&quot; &quot;GSM1523984&quot; &quot;GSM1523988&quot; &quot;GSM1524030&quot;</code></pre>
<p>Removing potential outlier samples</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="rna-data-preprocessing.html#cb64-1" tabindex="-1"></a>eset1 <span class="ot">&lt;-</span> eset[, <span class="sc">!</span><span class="fu">colnames</span>(eset)<span class="sc">%in%</span>res]</span></code></pre></div>
</div>
<div id="pca-analysis-of-molecular-subtypes" class="section level2 hasAnchor" number="3.7">
<h2><span class="header-section-number">3.7</span> PCA analysis of molecular subtypes<a href="rna-data-preprocessing.html#pca-analysis-of-molecular-subtypes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="rna-data-preprocessing.html#cb65-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;pdata_acrg&quot;</span>, <span class="at">package =</span> <span class="st">&quot;IOBR&quot;</span>)</span>
<span id="cb65-2"><a href="rna-data-preprocessing.html#cb65-2" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">iobr_pca</span>(<span class="at">data       =</span> eset1,</span>
<span id="cb65-3"><a href="rna-data-preprocessing.html#cb65-3" tabindex="-1"></a>              <span class="at">is.matrix   =</span> <span class="cn">TRUE</span>,</span>
<span id="cb65-4"><a href="rna-data-preprocessing.html#cb65-4" tabindex="-1"></a>              <span class="at">scale       =</span> <span class="cn">TRUE</span>,</span>
<span id="cb65-5"><a href="rna-data-preprocessing.html#cb65-5" tabindex="-1"></a>              <span class="at">is.log      =</span> <span class="cn">FALSE</span>,</span>
<span id="cb65-6"><a href="rna-data-preprocessing.html#cb65-6" tabindex="-1"></a>              <span class="at">pdata       =</span> pdata_acrg, </span>
<span id="cb65-7"><a href="rna-data-preprocessing.html#cb65-7" tabindex="-1"></a>              <span class="at">id_pdata    =</span> <span class="st">&quot;ID&quot;</span>, </span>
<span id="cb65-8"><a href="rna-data-preprocessing.html#cb65-8" tabindex="-1"></a>              <span class="at">group       =</span> <span class="st">&quot;Subtype&quot;</span>,</span>
<span id="cb65-9"><a href="rna-data-preprocessing.html#cb65-9" tabindex="-1"></a>              <span class="at">geom.ind    =</span> <span class="st">&quot;point&quot;</span>, </span>
<span id="cb65-10"><a href="rna-data-preprocessing.html#cb65-10" tabindex="-1"></a>              <span class="at">cols        =</span> <span class="st">&quot;normal&quot;</span>,</span>
<span id="cb65-11"><a href="rna-data-preprocessing.html#cb65-11" tabindex="-1"></a>              <span class="at">palette     =</span> <span class="st">&quot;jama&quot;</span>, </span>
<span id="cb65-12"><a href="rna-data-preprocessing.html#cb65-12" tabindex="-1"></a>              <span class="at">repel       =</span> <span class="cn">FALSE</span>,</span>
<span id="cb65-13"><a href="rna-data-preprocessing.html#cb65-13" tabindex="-1"></a>              <span class="at">ncp         =</span> <span class="dv">5</span>,</span>
<span id="cb65-14"><a href="rna-data-preprocessing.html#cb65-14" tabindex="-1"></a>              <span class="at">axes        =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>),</span>
<span id="cb65-15"><a href="rna-data-preprocessing.html#cb65-15" tabindex="-1"></a>              <span class="at">addEllipses =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## 
##       CIN       EBV       EMT        GS       MSI MSS/TP53- MSS/TP53+ 
##         0         0        42         0        68       106        79 
## [1] &quot;&gt;&gt;== colors for group: &quot;</code></pre>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="rna-data-preprocessing.html#cb67-1" tabindex="-1"></a>res</span></code></pre></div>
<p><img src="data-preprocessing_files/figure-html/unnamed-chunk-14-1.png" width="729.6" style="display: block; margin: auto;" /></p>
</div>
<div id="batch-effect-correction" class="section level2 hasAnchor" number="3.8">
<h2><span class="header-section-number">3.8</span> Batch effect correction<a href="rna-data-preprocessing.html#batch-effect-correction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="for-microarray-data" class="section level3 hasAnchor" number="3.8.1">
<h3><span class="header-section-number">3.8.1</span> For microarray data<a href="rna-data-preprocessing.html#for-microarray-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Obtaining another data set from GEO <a href="https://www.ncbi.nlm.nih.gov/pubmed/24935174/">Gastric cancer: GSE57303</a> using <code>GEOquery</code> R package.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="rna-data-preprocessing.html#cb68-1" tabindex="-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: This process may take a few minutes which depends on the internet connection speed. Please wait for its completion.</span></span>
<span id="cb68-2"><a href="rna-data-preprocessing.html#cb68-2" tabindex="-1"></a>eset_geo<span class="ot">&lt;-</span><span class="fu">getGEO</span>(<span class="at">GEO     =</span> <span class="st">&quot;GSE57303&quot;</span>, <span class="at">getGPL  =</span> F, <span class="at">destdir =</span> <span class="st">&quot;./&quot;</span>)</span>
<span id="cb68-3"><a href="rna-data-preprocessing.html#cb68-3" tabindex="-1"></a>eset2    <span class="ot">&lt;-</span>eset_geo[[<span class="dv">1</span>]]</span>
<span id="cb68-4"><a href="rna-data-preprocessing.html#cb68-4" tabindex="-1"></a>eset2    <span class="ot">&lt;-</span><span class="fu">exprs</span>(eset2)</span>
<span id="cb68-5"><a href="rna-data-preprocessing.html#cb68-5" tabindex="-1"></a>eset2[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##           GSM1379261 GSM1379262 GSM1379263 GSM1379264 GSM1379265
## 1007_s_at    8.34746    9.67994    8.62643    8.59301    8.63046
## 1053_at      5.07972    4.46377    5.29685    5.78983    4.33359
## 117_at       5.65558    4.48732    4.21615    5.47984    5.20816
## 121_at       5.95123    7.09056    6.19903    5.89872    5.91323
## 1255_g_at    1.66923    1.98758    1.73083    1.56687    1.63332</code></pre>
<p>Annotation of genes in the expression matrix and removal of duplicate genes.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="rna-data-preprocessing.html#cb70-1" tabindex="-1"></a>eset2<span class="ot">&lt;-</span><span class="fu">anno_eset</span>(<span class="at">eset       =</span> eset2,</span>
<span id="cb70-2"><a href="rna-data-preprocessing.html#cb70-2" tabindex="-1"></a>                 <span class="at">annotation =</span> anno_hug133plus2,</span>
<span id="cb70-3"><a href="rna-data-preprocessing.html#cb70-3" tabindex="-1"></a>                 <span class="at">symbol     =</span> <span class="st">&quot;symbol&quot;</span>,</span>
<span id="cb70-4"><a href="rna-data-preprocessing.html#cb70-4" tabindex="-1"></a>                 <span class="at">probe      =</span> <span class="st">&quot;probe_id&quot;</span>,</span>
<span id="cb70-5"><a href="rna-data-preprocessing.html#cb70-5" tabindex="-1"></a>                 <span class="at">method     =</span> <span class="st">&quot;mean&quot;</span>)</span>
<span id="cb70-6"><a href="rna-data-preprocessing.html#cb70-6" tabindex="-1"></a>eset2[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##         GSM1379261 GSM1379262 GSM1379263 GSM1379264 GSM1379265
## ND4        13.1695    13.1804    13.0600    12.4544    13.0457
## ATP6       13.1433    13.0814    13.0502    12.4831    13.1168
## SH3KBP1    12.9390    13.1620    12.9773    12.8745    13.1169
## COX2       13.0184    13.0489    12.8621    12.7489    12.9732
## RPL41      13.0201    12.6034    12.7929    13.0153    12.9404</code></pre>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="rna-data-preprocessing.html#cb72-1" tabindex="-1"></a>eset_com <span class="ot">&lt;-</span> <span class="fu">remove_batcheffect</span>( <span class="at">eset1       =</span> eset1,  </span>
<span id="cb72-2"><a href="rna-data-preprocessing.html#cb72-2" tabindex="-1"></a>                                <span class="at">eset2       =</span> eset2,   </span>
<span id="cb72-3"><a href="rna-data-preprocessing.html#cb72-3" tabindex="-1"></a>                                <span class="at">eset3       =</span> <span class="cn">NULL</span>,</span>
<span id="cb72-4"><a href="rna-data-preprocessing.html#cb72-4" tabindex="-1"></a>                                <span class="at">id_type     =</span> <span class="st">&quot;symbol&quot;</span>,</span>
<span id="cb72-5"><a href="rna-data-preprocessing.html#cb72-5" tabindex="-1"></a>                                <span class="at">data_type   =</span> <span class="st">&quot;array&quot;</span>, </span>
<span id="cb72-6"><a href="rna-data-preprocessing.html#cb72-6" tabindex="-1"></a>                                <span class="at">cols        =</span> <span class="st">&quot;normal&quot;</span>, </span>
<span id="cb72-7"><a href="rna-data-preprocessing.html#cb72-7" tabindex="-1"></a>                                <span class="at">palette     =</span> <span class="st">&quot;jama&quot;</span>, </span>
<span id="cb72-8"><a href="rna-data-preprocessing.html#cb72-8" tabindex="-1"></a>                                <span class="at">log2        =</span> <span class="cn">TRUE</span>, </span>
<span id="cb72-9"><a href="rna-data-preprocessing.html#cb72-9" tabindex="-1"></a>                                <span class="at">check_eset  =</span> <span class="cn">TRUE</span>,</span>
<span id="cb72-10"><a href="rna-data-preprocessing.html#cb72-10" tabindex="-1"></a>                                <span class="at">adjust_eset =</span> <span class="cn">TRUE</span>,</span>
<span id="cb72-11"><a href="rna-data-preprocessing.html#cb72-11" tabindex="-1"></a>                                <span class="at">repel       =</span> <span class="cn">FALSE</span>,</span>
<span id="cb72-12"><a href="rna-data-preprocessing.html#cb72-12" tabindex="-1"></a>                                <span class="at">path        =</span> <span class="st">&quot;result&quot;</span>)</span></code></pre></div>
<pre><code>## 
## eset1 eset2 
##   295    70 
## [1] &quot;&gt;&gt;== colors for group: &quot;
## 
## eset1 eset2 
##   295    70 
## [1] &quot;&gt;&gt;== colors for group: &quot;</code></pre>
<p><img src="data-preprocessing_files/figure-html/unnamed-chunk-17-1.png" width="1056" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="rna-data-preprocessing.html#cb74-1" tabindex="-1"></a><span class="fu">dim</span>(eset_com)</span></code></pre></div>
<pre><code>## [1] 21752   365</code></pre>
</div>
<div id="for-rnaseq-count-data" class="section level3 hasAnchor" number="3.8.2">
<h3><span class="header-section-number">3.8.2</span> For RNAseq count data<a href="rna-data-preprocessing.html#for-rnaseq-count-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="rna-data-preprocessing.html#cb76-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;eset_stad&quot;</span>, <span class="at">package =</span> <span class="st">&quot;IOBR&quot;</span>)</span>
<span id="cb76-2"><a href="rna-data-preprocessing.html#cb76-2" tabindex="-1"></a><span class="fu">head</span>(eset_stad)</span></code></pre></div>
<pre><code>##                 TCGA-BR-6455 TCGA-BR-7196 TCGA-BR-8371 TCGA-BR-8380
## ENSG00000000003         8006         2114          767         1556
## ENSG00000000005            1            0            5            5
## ENSG00000000419         3831         2600         1729         1760
## ENSG00000000457         1126          745         1040         1260
## ENSG00000000460          857          463          231          432
## ENSG00000000938          758         1126          557          557
##                 TCGA-BR-8592 TCGA-BR-8686 TCGA-BR-A4IV TCGA-BR-A4J4
## ENSG00000000003         2806         2923         1524         7208
## ENSG00000000005           60            1           22            2
## ENSG00000000419         2273         1934         2838         4418
## ENSG00000000457         1814          707         1683         1335
## ENSG00000000460          635          323          270          423
## ENSG00000000938          828          666          760          597
##                 TCGA-BR-A4J9 TCGA-FP-7916
## ENSG00000000003          711         2747
## ENSG00000000005            0            3
## ENSG00000000419         2426         2824
## ENSG00000000457         1590         1672
## ENSG00000000460          276          773
## ENSG00000000938          370          688</code></pre>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="rna-data-preprocessing.html#cb78-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;eset_blca&quot;</span>, <span class="at">package =</span> <span class="st">&quot;IOBR&quot;</span>)</span>
<span id="cb78-2"><a href="rna-data-preprocessing.html#cb78-2" tabindex="-1"></a><span class="fu">head</span>(eset_blca)</span></code></pre></div>
<pre><code>##                 TCGA-2F-A9KO TCGA-2F-A9KP TCGA-2F-A9KQ TCGA-2F-A9KR
## ENSG00000000003         6092        11652         5426         4383
## ENSG00000000005            0            4            1            1
## ENSG00000000419         3072         2656         1983         2061
## ENSG00000000457         1302          984         1134         1092
## ENSG00000000460          779          924          421          386
## ENSG00000000938          436          116          312          590
##                 TCGA-2F-A9KT
## ENSG00000000003         3334
## ENSG00000000005            0
## ENSG00000000419         2930
## ENSG00000000457          496
## ENSG00000000460          318
## ENSG00000000938          362</code></pre>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="rna-data-preprocessing.html#cb80-1" tabindex="-1"></a>eset_com <span class="ot">&lt;-</span> <span class="fu">remove_batcheffect</span>(eset_stad, eset_blca, <span class="at">id_type =</span> <span class="st">&quot;ensembl&quot;</span>, <span class="at">data_type =</span> <span class="st">&quot;count&quot;</span>)</span></code></pre></div>
<pre><code>## Found 2 batches
## Using null model in ComBat-seq.
## Adjusting for 0 covariate(s) or covariate level(s)
## Estimating dispersions
## Fitting the GLM model
## Shrinkage off - using GLM estimates for parameters
## Adjusting the data</code></pre>
<pre><code>## Warning in count2tpm(countMat = combined.expr.combat, idType = id_type, :
## &gt;&gt;&gt;--- Omit 1263 genes of which length is not available !</code></pre>
<pre><code>## 
## eset1 eset2 
##    10     5 
## [1] &quot;&gt;&gt;== colors for group: &quot;
## 
## eset1 eset2 
##    10     5 
## [1] &quot;&gt;&gt;== colors for group: &quot;
## 
## eset1 eset2 
##    10     5 
## [1] &quot;&gt;&gt;== colors for group: &quot;</code></pre>
<p><img src="data-preprocessing_files/figure-html/unnamed-chunk-18-1.png" width="1056" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="rna-data-preprocessing.html#cb84-1" tabindex="-1"></a><span class="co"># The returned matrix is the count matrix after removing the batches.</span></span>
<span id="cb84-2"><a href="rna-data-preprocessing.html#cb84-2" tabindex="-1"></a><span class="fu">head</span>(eset_com)</span></code></pre></div>
<pre><code>##                 TCGA-BR-6455 TCGA-BR-7196 TCGA-BR-8371 TCGA-BR-8380
## ENSG00000000003        10264         3536         1710         2964
## ENSG00000000005            1            0            4            5
## ENSG00000000419         4500         3099         2111         2167
## ENSG00000000457         1203          707         1106         1353
## ENSG00000000460         1059          590          310          560
## ENSG00000000938          731         1202          507          485
##                 TCGA-BR-8592 TCGA-BR-8686 TCGA-BR-A4IV TCGA-BR-A4J4
## ENSG00000000003         4761         3964         3115         9565
## ENSG00000000005           33            1           14            3
## ENSG00000000419         2782         2270         3444         5176
## ENSG00000000457         2089          817         1845         1469
## ENSG00000000460          810          405          368          548
## ENSG00000000938          769          723          677          532
##                 TCGA-BR-A4J9 TCGA-FP-7916 TCGA-2F-A9KO TCGA-2F-A9KP
## ENSG00000000003         1739         4371         2812         6796
## ENSG00000000005            0            3            0           10
## ENSG00000000419         2943         3362         2189         1849
## ENSG00000000457         1804         2044          994          817
## ENSG00000000460          371          959          495          584
## ENSG00000000938          281          654          456          156
##                 TCGA-2F-A9KQ TCGA-2F-A9KR TCGA-2F-A9KT
## ENSG00000000003         1971         1429         1057
## ENSG00000000005            1            1            0
## ENSG00000000419         1355         1420         2094
## ENSG00000000457          916          876          438
## ENSG00000000460          251          230          190
## ENSG00000000938          353          604          383</code></pre>
</div>
</div>
<div id="references" class="section level2 hasAnchor" number="3.9">
<h2><span class="header-section-number">3.9</span> References<a href="rna-data-preprocessing.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Wang et al., (2019). The UCSCXenaTools R package: a toolkit for accessing genomics data from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq. Journal of Open Source Software, 4(40), 1627, <a href="https://doi.org/10.21105/joss.01627" class="uri">https://doi.org/10.21105/joss.01627</a></p>
<p>Zhang et al., ComBat-seq: batch effect adjustment for RNA-seq count data, NAR Genomics and Bioinformatics, Volume 2, Issue 3, September 2020, lqaa078, <a href="https://doi.org/10.1093/nargab/lqaa078" class="uri">https://doi.org/10.1093/nargab/lqaa078</a></p>
<p>Leek, J. T., et al., (2012). The sva package for removing batch effects and other unwanted variation in high-throughput experiments. Bioinformatics, 28(6), 882-883.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="how-to-use-iobr.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="signature-score-calculation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/DongqiangZeng0808/data-preprocessing.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
