<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 TME and genomic interaction | IOBR (Immuno-Oncology Biological Research)</title>
  <meta name="description" content="The tutorial of IOBR package" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 TME and genomic interaction | IOBR (Immuno-Oncology Biological Research)" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://iobr.github.io/book/./fig/cover.png" />
  <meta property="og:description" content="The tutorial of IOBR package" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 TME and genomic interaction | IOBR (Immuno-Oncology Biological Research)" />
  
  <meta name="twitter:description" content="The tutorial of IOBR package" />
  <meta name="twitter:image" content="https://iobr.github.io/book/./fig/cover.png" />

<meta name="author" content="Dongqiang Zeng, Yiran Fang" />


<meta name="date" content="2024-01-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="tumor-ecosystem-analysis.html"/>
<link rel="next" href="tme-modeling.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The tutorial of IOBR</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><strong>Introduction</strong></a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#introduction-1"><i class="fa fa-check"></i><b>0.1</b> üìö Introduction</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>0.2</b> üíø License</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#previous-publication"><i class="fa fa-check"></i><b>0.3</b> üèÜ Previous publication</a></li>
<li class="chapter" data-level="0.4" data-path="index.html"><a href="index.html#major-updates"><i class="fa fa-check"></i><b>0.4</b> ‚è≥ Major Updates</a></li>
<li class="chapter" data-level="0.5" data-path="index.html"><a href="index.html#reporting-bugs"><i class="fa fa-check"></i><b>0.5</b> ‚úâÔ∏è Reporting bugs</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html"><i class="fa fa-check"></i><b>1</b> <strong>How to install IOBR</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#install-dependency-packages"><i class="fa fa-check"></i><b>1.1</b> ‚è≥ Install Dependency Packages</a></li>
<li class="chapter" data-level="1.2" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#install-iobr-package"><i class="fa fa-check"></i><b>1.2</b> üíª Install IOBR package</a></li>
<li class="chapter" data-level="1.3" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#how-to-update-iobr"><i class="fa fa-check"></i><b>1.3</b> üìç How to update IOBR</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="how-to-use-iobr.html"><a href="how-to-use-iobr.html"><i class="fa fa-check"></i><b>2</b> <strong>How to use IOBR</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-to-use-iobr.html"><a href="how-to-use-iobr.html#the-main-pipeline-of-iobr"><i class="fa fa-check"></i><b>2.1</b> üîç The main pipeline of IOBR</a></li>
<li class="chapter" data-level="2.2" data-path="how-to-use-iobr.html"><a href="how-to-use-iobr.html#main-functions-of-iobr"><i class="fa fa-check"></i><b>2.2</b> üóÇ Main Functions of IOBR</a></li>
<li class="chapter" data-level="2.3" data-path="how-to-use-iobr.html"><a href="how-to-use-iobr.html#current-working-environment"><i class="fa fa-check"></i><b>2.3</b> üåê Current working environment</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html"><i class="fa fa-check"></i><b>3</b> <strong>RNA Data preprocessing</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#loading-packages"><i class="fa fa-check"></i><b>3.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.2" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#download-array-data-using-geoquery"><i class="fa fa-check"></i><b>3.2</b> Download array data using <code>GEOquery</code></a></li>
<li class="chapter" data-level="3.3" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#gene-annotation"><i class="fa fa-check"></i><b>3.3</b> Gene Annotation</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#for-array-data-hgu133plus-2-affaymetrix"><i class="fa fa-check"></i><b>3.3.1</b> For Array data: HGU133PLUS-2 (Affaymetrix)</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#download-rnaseq-data-using-ucscxenatools"><i class="fa fa-check"></i><b>3.4</b> Download RNAseq data using <code>UCSCXenaTools</code></a></li>
<li class="chapter" data-level="3.5" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#normalization-and-gene-annotation"><i class="fa fa-check"></i><b>3.5</b> Normalization and Gene annotation</a></li>
<li class="chapter" data-level="3.6" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#identifying-outlier-samples"><i class="fa fa-check"></i><b>3.6</b> Identifying outlier samples</a></li>
<li class="chapter" data-level="3.7" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#pca-analysis-of-molecular-subtypes"><i class="fa fa-check"></i><b>3.7</b> PCA analysis of molecular subtypes</a></li>
<li class="chapter" data-level="3.8" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#batch-effect-correction"><i class="fa fa-check"></i><b>3.8</b> Batch effect correction</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#for-microarray-data"><i class="fa fa-check"></i><b>3.8.1</b> For microarray data</a></li>
<li class="chapter" data-level="3.8.2" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#for-rnaseq-count-data"><i class="fa fa-check"></i><b>3.8.2</b> For RNAseq count data</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#references"><i class="fa fa-check"></i><b>3.9</b> References</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html"><i class="fa fa-check"></i><b>4</b> <strong>Signature Score Calculation</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#loading-packages-1"><i class="fa fa-check"></i><b>4.1</b> Loading packages</a></li>
<li class="chapter" data-level="4.2" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#downloading-data-for-example"><i class="fa fa-check"></i><b>4.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="4.3" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#signature-score-estimation"><i class="fa fa-check"></i><b>4.3</b> Signature score estimation</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#signature-collection-of-iobr"><i class="fa fa-check"></i><b>4.3.1</b> Signature collection of IOBR</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#estimation-of-signature-using-pca-method"><i class="fa fa-check"></i><b>4.4</b> Estimation of signature using PCA method</a></li>
<li class="chapter" data-level="4.5" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#estimated-using-the-ssgsea-methodology"><i class="fa fa-check"></i><b>4.5</b> Estimated using the ssGSEA methodology</a></li>
<li class="chapter" data-level="4.6" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#calculated-using-the-z-score-function."><i class="fa fa-check"></i><b>4.6</b> Calculated using the z-score function.</a></li>
<li class="chapter" data-level="4.7" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#calculated-using-all-three-methods-at-the-same-time"><i class="fa fa-check"></i><b>4.7</b> Calculated using all three methods at the same time</a></li>
<li class="chapter" data-level="4.8" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#classification-of-signatures"><i class="fa fa-check"></i><b>4.8</b> Classification of signatures</a></li>
<li class="chapter" data-level="4.9" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#how-to-customise-the-signature-gene-list-for-calculate_signature_score"><i class="fa fa-check"></i><b>4.9</b> How to customise the signature gene list for <code>calculate_signature_score</code></a>
<ul>
<li class="chapter" data-level="4.9.1" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#method-1-use-excel-for-storage-and-construction"><i class="fa fa-check"></i><b>4.9.1</b> Method-1: Use excel for storage and construction</a></li>
<li class="chapter" data-level="4.9.2" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#method-2-build-the-list-structure-directly"><i class="fa fa-check"></i><b>4.9.2</b> Method-2: Build the list structure directly</a></li>
<li class="chapter" data-level="4.9.3" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#method3-add-the-new-signature-to-the-existing-gene-list"><i class="fa fa-check"></i><b>4.9.3</b> Method3: Add the new signature to the existing gene list</a></li>
</ul></li>
<li class="chapter" data-level="4.10" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#how-to-export-gene-signature"><i class="fa fa-check"></i><b>4.10</b> How to export gene signature</a></li>
<li class="chapter" data-level="4.11" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#references-1"><i class="fa fa-check"></i><b>4.11</b> References</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html"><i class="fa fa-check"></i><b>5</b> <strong>TME deconvolution</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#loading-packages-2"><i class="fa fa-check"></i><b>5.1</b> Loading packages</a></li>
<li class="chapter" data-level="5.2" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#downloading-data-for-example-1"><i class="fa fa-check"></i><b>5.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="5.3" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#available-methods-to-decode-tme-contexture"><i class="fa fa-check"></i><b>5.3</b> Available Methods to Decode TME Contexture</a></li>
<li class="chapter" data-level="5.4" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-1-cibersort"><i class="fa fa-check"></i><b>5.4</b> Method 1: CIBERSORT</a></li>
<li class="chapter" data-level="5.5" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-2-epic"><i class="fa fa-check"></i><b>5.5</b> Method 2: EPIC</a></li>
<li class="chapter" data-level="5.6" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-3-mcpcounter"><i class="fa fa-check"></i><b>5.6</b> Method 3: MCPcounter</a></li>
<li class="chapter" data-level="5.7" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-4-xcell"><i class="fa fa-check"></i><b>5.7</b> Method 4: xCELL</a></li>
<li class="chapter" data-level="5.8" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-5-estimate"><i class="fa fa-check"></i><b>5.8</b> Method 5: ESTIMATE</a></li>
<li class="chapter" data-level="5.9" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-6-timer"><i class="fa fa-check"></i><b>5.9</b> Method 6: TIMER</a></li>
<li class="chapter" data-level="5.10" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-7-quantiseq"><i class="fa fa-check"></i><b>5.10</b> Method 7: quanTIseq</a></li>
<li class="chapter" data-level="5.11" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-8-ips"><i class="fa fa-check"></i><b>5.11</b> Method 8: IPS</a></li>
<li class="chapter" data-level="5.12" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#combination-of-above-deconvolution-results"><i class="fa fa-check"></i><b>5.12</b> Combination of above deconvolution results</a></a></li>
<li class="chapter" data-level="5.13" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#how-to-customise-the-signature-matrix-for-svr-and-lesi-algorithm"><i class="fa fa-check"></i><b>5.13</b> How to customise the signature matrix for <code>SVR</code> and <code>lesi</code> algorithm</a></li>
<li class="chapter" data-level="5.14" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#references-2"><i class="fa fa-check"></i><b>5.14</b> References</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html"><i class="fa fa-check"></i><b>6</b> <strong>Signature Score and Relevant phenotypes</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#loading-packages-3"><i class="fa fa-check"></i><b>6.1</b> Loading packages</a></li>
<li class="chapter" data-level="6.2" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#downloading-data-for-example-2"><i class="fa fa-check"></i><b>6.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="6.3" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#gene-annotation-1"><i class="fa fa-check"></i><b>6.3</b> Gene Annotation</a></li>
<li class="chapter" data-level="6.4" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#estimation-of-signatures"><i class="fa fa-check"></i><b>6.4</b> Estimation of signatures</a></li>
<li class="chapter" data-level="6.5" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#combining-score-data-and-phenotype-data"><i class="fa fa-check"></i><b>6.5</b> Combining score data and phenotype data</a></li>
<li class="chapter" data-level="6.6" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#identifying-features-associated-with-survival"><i class="fa fa-check"></i><b>6.6</b> Identifying features associated with survival</a></li>
<li class="chapter" data-level="6.7" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#visulization-using-heatmap"><i class="fa fa-check"></i><b>6.7</b> Visulization using heatmap</a></li>
<li class="chapter" data-level="6.8" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#focus-on-target-signatures"><i class="fa fa-check"></i><b>6.8</b> Focus on target signatures</a></li>
<li class="chapter" data-level="6.9" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#survival-analysis-and-visulization"><i class="fa fa-check"></i><b>6.9</b> Survival analysis and visulization</a>
<ul>
<li class="chapter" data-level="6.9.1" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#kaplan-meier-plot"><i class="fa fa-check"></i><b>6.9.1</b> Kaplan-Meier plot</a></li>
<li class="chapter" data-level="6.9.2" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#time-dependent-roc-curve"><i class="fa fa-check"></i><b>6.9.2</b> Time-Dependent ROC curve</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#batch-correlation-analysis"><i class="fa fa-check"></i><b>6.10</b> Batch correlation analysis</a>
<ul>
<li class="chapter" data-level="6.10.1" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#finding-continuity-variables-associated-with-signatures"><i class="fa fa-check"></i><b>6.10.1</b> Finding continuity variables associated with signatures</a></li>
<li class="chapter" data-level="6.10.2" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#identifying-category-variables-linked-to-signatures"><i class="fa fa-check"></i><b>6.10.2</b> Identifying Category Variables Linked to Signatures</a></li>
<li class="chapter" data-level="6.10.3" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#for-multicategorical-variables-2-subgroups"><i class="fa fa-check"></i><b>6.10.3</b> For multicategorical variables (&gt;2 subgroups)</a></li>
</ul></li>
<li class="chapter" data-level="6.11" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#reference"><i class="fa fa-check"></i><b>6.11</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html"><i class="fa fa-check"></i><b>7</b> <strong>TME Interaction analysis</strong></a>
<ul>
<li class="chapter" data-level="7.1" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#loading-packages-4"><i class="fa fa-check"></i><b>7.1</b> Loading packages</a></li>
<li class="chapter" data-level="7.2" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#downloading-data-for-example-3"><i class="fa fa-check"></i><b>7.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="7.3" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#gene-annotation-hgu133plus-2-affaymetrix"><i class="fa fa-check"></i><b>7.3</b> Gene Annotation: HGU133PLUS-2 (Affaymetrix)</a></li>
<li class="chapter" data-level="7.4" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#tme-deconvolution-using-cibersort-algorithm"><i class="fa fa-check"></i><b>7.4</b> TME deconvolution using CIBERSORT algorithm</a></li>
<li class="chapter" data-level="7.5" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#identifying-tme-patterns"><i class="fa fa-check"></i><b>7.5</b> Identifying TME patterns</a></li>
<li class="chapter" data-level="7.6" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#cell-abundance-of-each-cluster"><i class="fa fa-check"></i><b>7.6</b> Cell abundance of each cluster</a></li>
<li class="chapter" data-level="7.7" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#deg-analysis-between-tme-subtypes"><i class="fa fa-check"></i><b>7.7</b> DEG analysis between TME subtypes</a></li>
<li class="chapter" data-level="7.8" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#identifying-signatures-associated-with-tme-clusters"><i class="fa fa-check"></i><b>7.8</b> Identifying signatures associated with TME clusters</a></li>
<li class="chapter" data-level="7.9" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#references-3"><i class="fa fa-check"></i><b>7.9</b> References</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html"><i class="fa fa-check"></i><b>8</b> <strong>Tumor ecosystem analysis</strong></a>
<ul>
<li class="chapter" data-level="8.1" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#loading-packages-5"><i class="fa fa-check"></i><b>8.1</b> Loading packages</a></li>
<li class="chapter" data-level="8.2" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#downloading-data-for-example-4"><i class="fa fa-check"></i><b>8.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="8.3" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#gene-annotation-hgu133plus-2-affaymetrix-1"><i class="fa fa-check"></i><b>8.3</b> Gene Annotation: HGU133PLUS-2 (Affaymetrix)</a></li>
<li class="chapter" data-level="8.4" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#determine-tme-subtype-of-gastric-cancer-using-tmeclassifier-r-package"><i class="fa fa-check"></i><b>8.4</b> Determine TME subtype of gastric cancer using TMEclassifier R package</a></li>
<li class="chapter" data-level="8.5" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#deg-analysis-method1"><i class="fa fa-check"></i><b>8.5</b> DEG analysis: method1</a></li>
<li class="chapter" data-level="8.6" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#gsea-analysis-based-on-differential-express-gene-analysis-results"><i class="fa fa-check"></i><b>8.6</b> GSEA analysis based on differential express gene analysis results</a></li>
<li class="chapter" data-level="8.7" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#deg-analysis-method2"><i class="fa fa-check"></i><b>8.7</b> DEG analysis: method2</a></li>
<li class="chapter" data-level="8.8" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#identifying-signatures-associated-with-tme-clusters-1"><i class="fa fa-check"></i><b>8.8</b> Identifying signatures associated with TME clusters</a></li>
<li class="chapter" data-level="8.9" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#references-4"><i class="fa fa-check"></i><b>8.9</b> References</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html"><i class="fa fa-check"></i><b>9</b> <strong>TME and genomic interaction</strong></a>
<ul>
<li class="chapter" data-level="9.1" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#loading-packages-6"><i class="fa fa-check"></i><b>9.1</b> Loading packages</a></li>
<li class="chapter" data-level="9.2" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#genomic-data-prepare"><i class="fa fa-check"></i><b>9.2</b> Genomic data prepare</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#using-tcgabiolinks-to-download-genomic-data"><i class="fa fa-check"></i><b>9.2.1</b> Using TCGAbiolinks to download genomic data</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#identifying-mutations-associated-with-tme"><i class="fa fa-check"></i><b>9.3</b> Identifying Mutations Associated with TME</a></li>
<li class="chapter" data-level="9.4" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#oncoprint-of-result"><i class="fa fa-check"></i><b>9.4</b> OncoPrint of result</a></li>
<li class="chapter" data-level="9.5" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#boxplot-of-top-10-mutated-genes"><i class="fa fa-check"></i><b>9.5</b> Boxplot of top 10 mutated genes</a></li>
<li class="chapter" data-level="9.6" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#other-methods-to-obtaind-genomic-data"><i class="fa fa-check"></i><b>9.6</b> Other methods to obtaind genomic data</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#using-tcgamutations"><i class="fa fa-check"></i><b>9.6.1</b> Using TCGAmutations</a></li>
<li class="chapter" data-level="9.6.2" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#using-maftools"><i class="fa fa-check"></i><b>9.6.2</b> Using maftools</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#references-5"><i class="fa fa-check"></i><b>9.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="tme-modeling.html"><a href="tme-modeling.html"><i class="fa fa-check"></i><b>10</b> <strong>TME Modeling</strong></a>
<ul>
<li class="chapter" data-level="10.1" data-path="tme-modeling.html"><a href="tme-modeling.html#loading-packages-7"><i class="fa fa-check"></i><b>10.1</b> Loading packages</a></li>
<li class="chapter" data-level="10.2" data-path="tme-modeling.html"><a href="tme-modeling.html#data-prepare"><i class="fa fa-check"></i><b>10.2</b> Data prepare</a></li>
<li class="chapter" data-level="10.3" data-path="tme-modeling.html"><a href="tme-modeling.html#input-data-overall-survival-prepare"><i class="fa fa-check"></i><b>10.3</b> Input data (overall survival) prepare</a></li>
<li class="chapter" data-level="10.4" data-path="tme-modeling.html"><a href="tme-modeling.html#constructing-survival-prediction-models"><i class="fa fa-check"></i><b>10.4</b> Constructing survival prediction models</a></li>
<li class="chapter" data-level="10.5" data-path="tme-modeling.html"><a href="tme-modeling.html#input-data-response-prepare"><i class="fa fa-check"></i><b>10.5</b> Input data (Response) prepare</a></li>
<li class="chapter" data-level="10.6" data-path="tme-modeling.html"><a href="tme-modeling.html#constructing-prediction-models-for-response"><i class="fa fa-check"></i><b>10.6</b> Constructing prediction models for response</a></li>
<li class="chapter" data-level="10.7" data-path="tme-modeling.html"><a href="tme-modeling.html#references-6"><i class="fa fa-check"></i><b>10.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html"><i class="fa fa-check"></i><b>11</b> <strong>Some tips about IOBR</strong></a>
<ul>
<li class="chapter" data-level="11.1" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html#loading-packages-8"><i class="fa fa-check"></i><b>11.1</b> Loading packages</a></li>
<li class="chapter" data-level="11.2" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html#colour-configuration"><i class="fa fa-check"></i><b>11.2</b> Colour Configuration</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html#categorical-variable"><i class="fa fa-check"></i><b>11.2.1</b> categorical variable</a></li>
<li class="chapter" data-level="11.2.2" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html#higher-number-of-subgroups"><i class="fa fa-check"></i><b>11.2.2</b> Higher number of subgroups</a></li>
<li class="chapter" data-level="11.2.3" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html#gradient-colour-or-heatmap-colour-scheme"><i class="fa fa-check"></i><b>11.2.3</b> Gradient colour or heatmap colour scheme</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="references-7.html"><a href="references-7.html"><i class="fa fa-check"></i><b>12</b> <strong>References</strong></a>
<ul>
<li class="chapter" data-level="12.1" data-path="references-7.html"><a href="references-7.html#tme-deconvolution-1"><i class="fa fa-check"></i><b>12.1</b> TME deconvolution</a></li>
<li class="chapter" data-level="12.2" data-path="references-7.html"><a href="references-7.html#tme-signatures"><i class="fa fa-check"></i><b>12.2</b> TME Signatures</a></li>
<li class="chapter" data-level="12.3" data-path="references-7.html"><a href="references-7.html#data-sets"><i class="fa fa-check"></i><b>12.3</b> Data sets</a></li>
<li class="chapter" data-level="12.4" data-path="references-7.html"><a href="references-7.html#others"><i class="fa fa-check"></i><b>12.4</b> Others</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/DongqiangZeng0808" target="blank">Welcome to my GitHub</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">IOBR (Immuno-Oncology Biological Research)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="tme-and-genomic-interaction" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> <strong>TME and genomic interaction</strong><a href="tme-and-genomic-interaction.html#tme-and-genomic-interaction" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="loading-packages-6" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Loading packages<a href="tme-and-genomic-interaction.html#loading-packages-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb364"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb364-1"><a href="tme-and-genomic-interaction.html#cb364-1" tabindex="-1"></a><span class="fu">library</span>(IOBR)</span></code></pre></div>
</div>
<div id="genomic-data-prepare" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Genomic data prepare<a href="tme-and-genomic-interaction.html#genomic-data-prepare" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this section, we are going to use the MAF data of TCGA-STAD cohort as an example dataset. This dataset could be found in multiple places, here we show two ways to get it.</p>
<div id="using-tcgabiolinks-to-download-genomic-data" class="section level3 hasAnchor" number="9.2.1">
<h3><span class="header-section-number">9.2.1</span> Using TCGAbiolinks to download genomic data<a href="tme-and-genomic-interaction.html#using-tcgabiolinks-to-download-genomic-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>R Bioconductor package <strong><a href="https://bioconductor.org/packages/TCGAbiolinks/">TCGAbiolinks</a></strong> provides an R interface of <a href="https://portal.gdc.cancer.gov/">GDC data portal</a>, which stores updating TCGA data. You can check and install this package with the following code.</p>
<div class="sourceCode" id="cb365"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb365-1"><a href="tme-and-genomic-interaction.html#cb365-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;TCGAbiolinks&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb365-2"><a href="tme-and-genomic-interaction.html#cb365-2" tabindex="-1"></a>  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;TCGAbiolinks&quot;</span>)</span></code></pre></div>
<p>Then you can query, download and prepare the required dataset.</p>
<div class="sourceCode" id="cb366"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb366-1"><a href="tme-and-genomic-interaction.html#cb366-1" tabindex="-1"></a><span class="fu">library</span>(TCGAbiolinks)</span>
<span id="cb366-2"><a href="tme-and-genomic-interaction.html#cb366-2" tabindex="-1"></a>query <span class="ot">&lt;-</span> <span class="fu">GDCquery</span>(</span>
<span id="cb366-3"><a href="tme-and-genomic-interaction.html#cb366-3" tabindex="-1"></a>  <span class="at">project =</span> <span class="st">&quot;TCGA-STAD&quot;</span>, </span>
<span id="cb366-4"><a href="tme-and-genomic-interaction.html#cb366-4" tabindex="-1"></a>  <span class="at">data.category =</span> <span class="st">&quot;Simple Nucleotide Variation&quot;</span>, </span>
<span id="cb366-5"><a href="tme-and-genomic-interaction.html#cb366-5" tabindex="-1"></a>  <span class="at">access =</span> <span class="st">&quot;open&quot;</span>,</span>
<span id="cb366-6"><a href="tme-and-genomic-interaction.html#cb366-6" tabindex="-1"></a>  <span class="at">data.type =</span> <span class="st">&quot;Masked Somatic Mutation&quot;</span>, </span>
<span id="cb366-7"><a href="tme-and-genomic-interaction.html#cb366-7" tabindex="-1"></a>  <span class="at">workflow.type =</span> <span class="st">&quot;Aliquot Ensemble Somatic Variant Merging and Masking&quot;</span></span>
<span id="cb366-8"><a href="tme-and-genomic-interaction.html#cb366-8" tabindex="-1"></a>)</span>
<span id="cb366-9"><a href="tme-and-genomic-interaction.html#cb366-9" tabindex="-1"></a><span class="fu">GDCdownload</span>(query)</span>
<span id="cb366-10"><a href="tme-and-genomic-interaction.html#cb366-10" tabindex="-1"></a>maf <span class="ot">&lt;-</span> <span class="fu">GDCprepare</span>(query)</span></code></pre></div>
<p>This <code>maf</code> object is a <code>data.frame</code>, you can use <code>read.maf()</code> from R package <strong><a href="https://github.com/PoisonAlien/maftools">Maftools</a></strong> to convert it to a MAF object.</p>
<p>In this example, we used the maf file of TCGA-STAD to extract the SNPs in it,and then transformed it into a non-negative matrix.</p>
<div class="sourceCode" id="cb367"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb367-1"><a href="tme-and-genomic-interaction.html#cb367-1" tabindex="-1"></a>(<span class="fu">load</span>(<span class="st">&quot;TCGA-STAD.maf.RData&quot;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;maf&quot;</code></pre>
<div class="sourceCode" id="cb369"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb369-1"><a href="tme-and-genomic-interaction.html#cb369-1" tabindex="-1"></a>mut_list <span class="ot">&lt;-</span> <span class="fu">make_mut_matrix</span>(<span class="at">maf =</span> maf, <span class="at">isTCGA   =</span> T, <span class="at">category =</span> <span class="st">&quot;multi&quot;</span>)</span></code></pre></div>
<pre><code>## -Validating
## -Silent variants: 45460 
## -Summarizing
## --Possible FLAGS among top ten genes:
##   TTN
##   MUC16
##   SYNE1
##   FLG
## -Processing clinical data
## --Missing clinical data
## -Finished in 11.4s elapsed (10.4s cpu) 
##        Frame_Shift_Del        Frame_Shift_Ins           In_Frame_Del 
##                  21547                   4526                   1196 
##           In_Frame_Ins      Missense_Mutation      Nonsense_Mutation 
##                    106                 102137                   5669 
##       Nonstop_Mutation            Splice_Site Translation_Start_Site 
##                    117                   2242                    107 
##    DEL    INS    ONP    SNP 
##  22997   4675     10 109965</code></pre>
<div class="sourceCode" id="cb371"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb371-1"><a href="tme-and-genomic-interaction.html#cb371-1" tabindex="-1"></a>mut <span class="ot">&lt;-</span> mut_list<span class="sc">$</span>snp</span></code></pre></div>
</div>
</div>
<div id="identifying-mutations-associated-with-tme" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Identifying Mutations Associated with TME<a href="tme-and-genomic-interaction.html#identifying-mutations-associated-with-tme" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The microenvironmental data from the TCGA-STAD expression matrix was merged. The Cuzick or Wilcoxon test was used to identify genetic variants associated with microenvironmental factors. CD_8_T_effector was used as the target variable in this example.</p>
<div class="sourceCode" id="cb372"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb372-1"><a href="tme-and-genomic-interaction.html#cb372-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;tcga_stad_sig&quot;</span>, <span class="at">package =</span> <span class="st">&quot;IOBR&quot;</span>)</span>
<span id="cb372-2"><a href="tme-and-genomic-interaction.html#cb372-2" tabindex="-1"></a>res<span class="ot">&lt;-</span><span class="fu">find_mutations</span>(<span class="at">mutation_matrix     =</span> mut, </span>
<span id="cb372-3"><a href="tme-and-genomic-interaction.html#cb372-3" tabindex="-1"></a>                    <span class="at">signature_matrix    =</span> tcga_stad_sig, </span>
<span id="cb372-4"><a href="tme-and-genomic-interaction.html#cb372-4" tabindex="-1"></a>                    <span class="at">id_signature_matrix =</span> <span class="st">&quot;ID&quot;</span>, </span>
<span id="cb372-5"><a href="tme-and-genomic-interaction.html#cb372-5" tabindex="-1"></a>                    <span class="at">signature           =</span> <span class="st">&quot;CD_8_T_effector&quot;</span>,</span>
<span id="cb372-6"><a href="tme-and-genomic-interaction.html#cb372-6" tabindex="-1"></a>                    <span class="at">palette             =</span> <span class="st">&quot;nrc&quot;</span>,</span>
<span id="cb372-7"><a href="tme-and-genomic-interaction.html#cb372-7" tabindex="-1"></a>                    <span class="at">min_mut_freq        =</span> <span class="fl">0.01</span>, </span>
<span id="cb372-8"><a href="tme-and-genomic-interaction.html#cb372-8" tabindex="-1"></a>                    <span class="at">plot                =</span> <span class="cn">TRUE</span>, </span>
<span id="cb372-9"><a href="tme-and-genomic-interaction.html#cb372-9" tabindex="-1"></a>                    <span class="at">jitter              =</span> <span class="cn">TRUE</span>, </span>
<span id="cb372-10"><a href="tme-and-genomic-interaction.html#cb372-10" tabindex="-1"></a>                    <span class="at">point.alpha         =</span> <span class="fl">0.25</span>)</span></code></pre></div>
<pre><code>## [1] &quot;&gt;&gt;&gt;&gt; Result of Cuzick Test&quot;
##             p.value  names statistic adjust_pvalue
## PIK3CA 6.279087e-10 PIK3CA  6.183261  3.139543e-07
## PLXNA4 8.196539e-05 PLXNA4  3.938580  2.049135e-02
## DMD    3.514677e-04    DMD  3.574075  3.882570e-02
## SPEG   3.665757e-04   SPEG  3.563047  3.882570e-02
## AHNAK2 3.882570e-04 AHNAK2  3.547940  3.882570e-02
## TCHH   6.571311e-04   TCHH  3.406867  5.476093e-02
## ABCC9  1.043516e-03  ABCC9  3.278524  7.453684e-02
## ANK3   1.470480e-03   ANK3  3.180447  7.713409e-02
## WDFY3  1.690248e-03  WDFY3  3.139867  7.713409e-02
## LRP1   1.762002e-03   LRP1  3.127666  7.713409e-02</code></pre>
<pre><code>## [1] &quot;&gt;&gt;&gt; Result of Wilcoxon test (top 10)&quot;
##             p.value  names statistic adjust_pvalue
## PIK3CA 3.045532e-11 PIK3CA      4006  1.522766e-08
## TCHH   2.585594e-05   TCHH      2929  6.463985e-03
## PLXNA4 5.257043e-05 PLXNA4      2904  7.754311e-03
## LRP1   6.203449e-05   LRP1      2330  7.754311e-03
## RNF213 1.066555e-04 RNF213      3827  1.066555e-02
## SPEG   1.809850e-04   SPEG      1854  1.439761e-02
## WDFY3  2.420595e-04  WDFY3      2626  1.439761e-02
## DMD    2.430706e-04    DMD      4995  1.439761e-02
## ANK3   2.591569e-04   ANK3      3845  1.439761e-02
## AHNAK2 3.456706e-04 AHNAK2      4828  1.728353e-02</code></pre>
<pre><code>## All mutation types: mut.</code></pre>
<pre><code>## Warning: You defined `cell_fun` for a heatmap with more than 100 rows or
## columns, which might be very slow to draw. Consider to use the
## vectorized version `layer_fun`.</code></pre>
<pre><code>## All mutation types: mut.</code></pre>
<pre><code>## Warning: You defined `cell_fun` for a heatmap with more than 100 rows or
## columns, which might be very slow to draw. Consider to use the
## vectorized version `layer_fun`.</code></pre>
</div>
<div id="oncoprint-of-result" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> OncoPrint of result<a href="tme-and-genomic-interaction.html#oncoprint-of-result" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-6"></span>
<img src="fig/0-OncoPrint-CD_8_T_effector.png" alt="OncoPrint" width="95%" />
<p class="caption">
Figure 9.1: OncoPrint
</p>
</div>
</div>
<div id="boxplot-of-top-10-mutated-genes" class="section level2 hasAnchor" number="9.5">
<h2><span class="header-section-number">9.5</span> Boxplot of top 10 mutated genes<a href="tme-and-genomic-interaction.html#boxplot-of-top-10-mutated-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-7"></span>
<img src="fig/4-Relevant_mutations_binary.png" alt="Top 10 mutated genes" width="95%" />
<p class="caption">
Figure 9.2: Top 10 mutated genes
</p>
</div>
</div>
<div id="other-methods-to-obtaind-genomic-data" class="section level2 hasAnchor" number="9.6">
<h2><span class="header-section-number">9.6</span> Other methods to obtaind genomic data<a href="tme-and-genomic-interaction.html#other-methods-to-obtaind-genomic-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="using-tcgamutations" class="section level3 hasAnchor" number="9.6.1">
<h3><span class="header-section-number">9.6.1</span> Using TCGAmutations<a href="tme-and-genomic-interaction.html#using-tcgamutations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>As its name, the R package <strong><a href="https://github.com/PoisonAlien/TCGAmutations">TCGAmutations</a></strong> provides pre-compiled, curated somatic mutations from 33 TCGA cohorts along with relevant clinical information for all sequenced samples. You can install it similar to the <strong>TCGAbiolinks</strong>.</p>
<div class="sourceCode" id="cb379"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb379-1"><a href="tme-and-genomic-interaction.html#cb379-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;TCGAmutations&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb379-2"><a href="tme-and-genomic-interaction.html#cb379-2" tabindex="-1"></a>  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;PoisonAlien/TCGAmutations&quot;</span>)</span></code></pre></div>
<p>It‚Äôs quite simple to use:</p>
<div class="sourceCode" id="cb380"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb380-1"><a href="tme-and-genomic-interaction.html#cb380-1" tabindex="-1"></a>maf <span class="ot">=</span> TCGAmutations<span class="sc">::</span><span class="fu">tcga_load</span>(<span class="at">study =</span> <span class="st">&quot;STAD&quot;</span>)</span>
<span id="cb380-2"><a href="tme-and-genomic-interaction.html#cb380-2" tabindex="-1"></a><span class="co"># Change `source` argument to Firehose for MAF files from Broad Firehose</span></span>
<span id="cb380-3"><a href="tme-and-genomic-interaction.html#cb380-3" tabindex="-1"></a><span class="co"># maf = TCGAmutations::tcga_load(study = &quot;STAD&quot;, source = &quot;Firehose&quot;)</span></span></code></pre></div>
</div>
<div id="using-maftools" class="section level3 hasAnchor" number="9.6.2">
<h3><span class="header-section-number">9.6.2</span> Using maftools<a href="tme-and-genomic-interaction.html#using-maftools" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>If you are a user of R package <strong><a href="https://github.com/PoisonAlien/maftools">Maftools</a></strong>, you can access and load the data in a similar way (because the author of <strong>TCGAmutations</strong> and <strong>Maftools</strong> is the same person).</p>
<div class="sourceCode" id="cb381"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb381-1"><a href="tme-and-genomic-interaction.html#cb381-1" tabindex="-1"></a><span class="co"># The following github can be changed to gitee</span></span>
<span id="cb381-2"><a href="tme-and-genomic-interaction.html#cb381-2" tabindex="-1"></a><span class="co"># it maybe fast in China mainland</span></span>
<span id="cb381-3"><a href="tme-and-genomic-interaction.html#cb381-3" tabindex="-1"></a>maftools<span class="sc">::</span><span class="fu">tcgaAvailable</span>(<span class="at">repo =</span> <span class="st">&quot;github&quot;</span>)</span>
<span id="cb381-4"><a href="tme-and-genomic-interaction.html#cb381-4" tabindex="-1"></a>maftools<span class="sc">::</span><span class="fu">tcgaLoad</span>(<span class="st">&quot;STAD&quot;</span>, <span class="at">repo =</span> <span class="st">&quot;github&quot;</span>)</span></code></pre></div>
<p>MAF data transformation
To prepare the data for the downstream analysis. We need to extract the SNV data in it and transform it into a non-negative matrix.</p>
<div class="sourceCode" id="cb382"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb382-1"><a href="tme-and-genomic-interaction.html#cb382-1" tabindex="-1"></a>mut_list <span class="ot">&lt;-</span> <span class="fu">make_mut_matrix</span>(<span class="at">maf =</span> maf, <span class="at">isTCGA =</span> <span class="cn">TRUE</span>, <span class="at">category =</span> <span class="st">&quot;multi&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="references-5" class="section level2 hasAnchor" number="9.7">
<h2><span class="header-section-number">9.7</span> References<a href="tme-and-genomic-interaction.html#references-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Wang et al., (2019). The UCSCXenaTools R package: a toolkit for accessing genomics data from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq. Journal of Open Source Software, 4(40), 1627, <a href="https://doi.org/10.21105/joss.01627" class="uri">https://doi.org/10.21105/joss.01627</a></p>
<p>Gu, Z. (2022) Complex Heatmap Visualization. iMeta.</p>
<p>Anand Mayakonda et al., (2018) Maftools: efficient and comprehensive analysis of somatic variants in cancer. Genome Research</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="tumor-ecosystem-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="tme-modeling.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/DongqiangZeng0808/tme-genomic-interaction.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
