<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Tumor ecosystem analysis | IOBR (Immuno-Oncology Biological Research)</title>
  <meta name="description" content="The tutorial of IOBR package" />
  <meta name="generator" content="bookdown 0.41.1 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Tumor ecosystem analysis | IOBR (Immuno-Oncology Biological Research)" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://iobr.github.io/book/./fig/cover.png" />
  <meta property="og:description" content="The tutorial of IOBR package" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Tumor ecosystem analysis | IOBR (Immuno-Oncology Biological Research)" />
  
  <meta name="twitter:description" content="The tutorial of IOBR package" />
  <meta name="twitter:image" content="https://iobr.github.io/book/./fig/cover.png" />

<meta name="author" content="Dongqiang Zeng, Yiran Fang" />


<meta name="date" content="2025-05-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="tme-interaction-analysis.html"/>
<link rel="next" href="tme-and-genomic-interaction.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The tutorial of IOBR</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><strong>Introduction</strong></a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#introduction-1"><i class="fa fa-check"></i><b>0.1</b> üìö Introduction</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>0.2</b> üíø License</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#citations"><i class="fa fa-check"></i><b>0.3</b> üèÜ Citations</a></li>
<li class="chapter" data-level="0.4" data-path="index.html"><a href="index.html#major-updates"><i class="fa fa-check"></i><b>0.4</b> ‚è≥ Major Updates</a></li>
<li class="chapter" data-level="0.5" data-path="index.html"><a href="index.html#reporting-bugs"><i class="fa fa-check"></i><b>0.5</b> ‚úâÔ∏è Reporting bugs</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html"><i class="fa fa-check"></i><b>1</b> <strong>How to install IOBR</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#install-dependency-packages"><i class="fa fa-check"></i><b>1.1</b> ‚è≥ Install Dependency Packages</a></li>
<li class="chapter" data-level="1.2" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#install-iobr-package"><i class="fa fa-check"></i><b>1.2</b> üíª Install IOBR package</a></li>
<li class="chapter" data-level="1.3" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#how-to-update-iobr"><i class="fa fa-check"></i><b>1.3</b> üìå How to update IOBR</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="how-to-use-iobr.html"><a href="how-to-use-iobr.html"><i class="fa fa-check"></i><b>2</b> <strong>How to use IOBR</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-to-use-iobr.html"><a href="how-to-use-iobr.html#the-main-pipeline-of-iobr"><i class="fa fa-check"></i><b>2.1</b> üîç The main pipeline of IOBR</a></li>
<li class="chapter" data-level="2.2" data-path="how-to-use-iobr.html"><a href="how-to-use-iobr.html#main-functions-of-iobr"><i class="fa fa-check"></i><b>2.2</b> üëÜ Main Functions of IOBR</a></li>
<li class="chapter" data-level="2.3" data-path="how-to-use-iobr.html"><a href="how-to-use-iobr.html#current-working-environment"><i class="fa fa-check"></i><b>2.3</b> üåê Current working environment</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html"><i class="fa fa-check"></i><b>3</b> <strong>RNA Data preprocessing</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#introduction-2"><i class="fa fa-check"></i><b>3.1</b> <strong>Introduction</strong></a></li>
<li class="chapter" data-level="3.2" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#loading-packages"><i class="fa fa-check"></i><b>3.2</b> Loading packages</a></li>
<li class="chapter" data-level="3.3" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#gene-annotation"><i class="fa fa-check"></i><b>3.3</b> Gene Annotation</a></li>
<li class="chapter" data-level="3.4" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#download-array-data-using-geoquery"><i class="fa fa-check"></i><b>3.4</b> Download array data using <code>GEOquery</code></a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#for-array-data-hgu133plus-2-affaymetrix"><i class="fa fa-check"></i><b>3.4.1</b> For Array data: HGU133PLUS-2 (Affaymetrix)</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#download-rnaseq-data-using-ucscxenatools"><i class="fa fa-check"></i><b>3.5</b> Download RNAseq data using <code>UCSCXenaTools</code></a></li>
<li class="chapter" data-level="3.6" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#normalization-and-gene-annotation"><i class="fa fa-check"></i><b>3.6</b> Normalization and Gene annotation</a></li>
<li class="chapter" data-level="3.7" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#identifying-outlier-samples"><i class="fa fa-check"></i><b>3.7</b> Identifying outlier samples</a></li>
<li class="chapter" data-level="3.8" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#pca-analysis-of-molecular-subtypes"><i class="fa fa-check"></i><b>3.8</b> PCA analysis of molecular subtypes</a></li>
<li class="chapter" data-level="3.9" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#batch-effect-correction"><i class="fa fa-check"></i><b>3.9</b> Batch effect correction</a>
<ul>
<li class="chapter" data-level="3.9.1" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#for-microarray-data"><i class="fa fa-check"></i><b>3.9.1</b> For microarray data</a></li>
<li class="chapter" data-level="3.9.2" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#for-rnaseq-count-data"><i class="fa fa-check"></i><b>3.9.2</b> For RNAseq count data</a></li>
</ul></li>
<li class="chapter" data-level="3.10" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#references"><i class="fa fa-check"></i><b>3.10</b> References</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html"><i class="fa fa-check"></i><b>4</b> <strong>Signature Score Calculation</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#loading-packages-1"><i class="fa fa-check"></i><b>4.1</b> Loading packages</a></li>
<li class="chapter" data-level="4.2" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#downloading-data-for-example"><i class="fa fa-check"></i><b>4.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="4.3" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#signature-score-estimation"><i class="fa fa-check"></i><b>4.3</b> Signature score estimation</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#signature-collection-of-iobr"><i class="fa fa-check"></i><b>4.3.1</b> Signature collection of IOBR</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#estimation-of-signature-using-pca-method"><i class="fa fa-check"></i><b>4.4</b> Estimation of signature using PCA method</a></li>
<li class="chapter" data-level="4.5" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#estimated-using-the-ssgsea-methodology"><i class="fa fa-check"></i><b>4.5</b> Estimated using the ssGSEA methodology</a></li>
<li class="chapter" data-level="4.6" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#calculated-using-the-z-score-function."><i class="fa fa-check"></i><b>4.6</b> Calculated using the z-score function.</a></li>
<li class="chapter" data-level="4.7" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#calculated-using-all-three-methods-at-the-same-time"><i class="fa fa-check"></i><b>4.7</b> Calculated using all three methods at the same time</a></li>
<li class="chapter" data-level="4.8" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#classification-of-signatures"><i class="fa fa-check"></i><b>4.8</b> Classification of signatures</a></li>
<li class="chapter" data-level="4.9" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#how-to-customise-the-signature-gene-list-for-calculate_signature_score"><i class="fa fa-check"></i><b>4.9</b> How to customise the signature gene list for <code>calculate_signature_score</code></a>
<ul>
<li class="chapter" data-level="4.9.1" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#method-1-use-excel-for-storage-and-construction"><i class="fa fa-check"></i><b>4.9.1</b> Method-1: Use excel for storage and construction</a></li>
<li class="chapter" data-level="4.9.2" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#method-2-build-the-list-structure-directly"><i class="fa fa-check"></i><b>4.9.2</b> Method-2: Build the list structure directly</a></li>
<li class="chapter" data-level="4.9.3" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#method-3-add-the-new-signature-to-the-existing-gene-list"><i class="fa fa-check"></i><b>4.9.3</b> Method-3: Add the new signature to the existing gene list</a></li>
<li class="chapter" data-level="4.9.4" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#method-4-construct-cell-specific-gene-signatures-from-single-cell-differential-analysis-results"><i class="fa fa-check"></i><b>4.9.4</b> Method-4: Construct cell-specific gene signatures from single-cell differential analysis results</a></li>
</ul></li>
<li class="chapter" data-level="4.10" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#how-to-export-gene-signature"><i class="fa fa-check"></i><b>4.10</b> How to export gene signature</a></li>
<li class="chapter" data-level="4.11" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#references-1"><i class="fa fa-check"></i><b>4.11</b> References</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html"><i class="fa fa-check"></i><b>5</b> <strong>TME deconvolution</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#loading-packages-2"><i class="fa fa-check"></i><b>5.1</b> Loading packages</a></li>
<li class="chapter" data-level="5.2" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#downloading-data-for-example-1"><i class="fa fa-check"></i><b>5.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="5.3" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#available-methods-to-decode-tme-contexture"><i class="fa fa-check"></i><b>5.3</b> Available Methods to Decode TME Contexture</a></li>
<li class="chapter" data-level="5.4" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-1-cibersort"><i class="fa fa-check"></i><b>5.4</b> Method 1: CIBERSORT</a></li>
<li class="chapter" data-level="5.5" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-2-epic"><i class="fa fa-check"></i><b>5.5</b> Method 2: EPIC</a></li>
<li class="chapter" data-level="5.6" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-3-mcpcounter"><i class="fa fa-check"></i><b>5.6</b> Method 3: MCPcounter</a></li>
<li class="chapter" data-level="5.7" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-4-xcell"><i class="fa fa-check"></i><b>5.7</b> Method 4: xCELL</a></li>
<li class="chapter" data-level="5.8" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-5-estimate"><i class="fa fa-check"></i><b>5.8</b> Method 5: ESTIMATE</a></li>
<li class="chapter" data-level="5.9" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-6-timer"><i class="fa fa-check"></i><b>5.9</b> Method 6: TIMER</a></li>
<li class="chapter" data-level="5.10" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-7-quantiseq"><i class="fa fa-check"></i><b>5.10</b> Method 7: quanTIseq</a></li>
<li class="chapter" data-level="5.11" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-8-ips"><i class="fa fa-check"></i><b>5.11</b> Method 8: IPS</a></li>
<li class="chapter" data-level="5.12" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#combination-of-above-deconvolution-results"><i class="fa fa-check"></i><b>5.12</b> Combination of above deconvolution results</a></a></li>
<li class="chapter" data-level="5.13" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#how-to-customise-the-signature-matrix-for-svr-and-lesi-algorithm"><i class="fa fa-check"></i><b>5.13</b> How to customise the signature matrix for <code>SVR</code> and <code>lesi</code> algorithm</a></li>
<li class="chapter" data-level="5.14" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#references-2"><i class="fa fa-check"></i><b>5.14</b> References</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html"><i class="fa fa-check"></i><b>6</b> <strong>Signature Score and Relevant phenotypes</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#loading-packages-3"><i class="fa fa-check"></i><b>6.1</b> Loading packages</a></li>
<li class="chapter" data-level="6.2" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#downloading-data-for-example-2"><i class="fa fa-check"></i><b>6.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="6.3" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#gene-annotation-1"><i class="fa fa-check"></i><b>6.3</b> Gene Annotation</a></li>
<li class="chapter" data-level="6.4" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#estimation-of-signatures"><i class="fa fa-check"></i><b>6.4</b> Estimation of signatures</a></li>
<li class="chapter" data-level="6.5" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#combining-score-data-and-phenotype-data"><i class="fa fa-check"></i><b>6.5</b> Combining score data and phenotype data</a></li>
<li class="chapter" data-level="6.6" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#identifying-features-associated-with-survival"><i class="fa fa-check"></i><b>6.6</b> Identifying features associated with survival</a></li>
<li class="chapter" data-level="6.7" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#visulization-using-heatmap"><i class="fa fa-check"></i><b>6.7</b> Visulization using heatmap</a></li>
<li class="chapter" data-level="6.8" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#focus-on-target-signatures"><i class="fa fa-check"></i><b>6.8</b> Focus on target signatures</a></li>
<li class="chapter" data-level="6.9" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#survival-analysis-and-visulization"><i class="fa fa-check"></i><b>6.9</b> Survival analysis and visulization</a>
<ul>
<li class="chapter" data-level="6.9.1" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#kaplan-meier-plot"><i class="fa fa-check"></i><b>6.9.1</b> Kaplan-Meier plot</a></li>
<li class="chapter" data-level="6.9.2" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#time-dependent-roc-curve"><i class="fa fa-check"></i><b>6.9.2</b> Time-Dependent ROC curve</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#batch-correlation-analysis"><i class="fa fa-check"></i><b>6.10</b> Batch correlation analysis</a>
<ul>
<li class="chapter" data-level="6.10.1" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#finding-continuity-variables-associated-with-signatures"><i class="fa fa-check"></i><b>6.10.1</b> Finding continuity variables associated with signatures</a></li>
<li class="chapter" data-level="6.10.2" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#identifying-category-variables-linked-to-signatures"><i class="fa fa-check"></i><b>6.10.2</b> Identifying Category Variables Linked to Signatures</a></li>
<li class="chapter" data-level="6.10.3" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#for-multicategorical-variables-2-subgroups"><i class="fa fa-check"></i><b>6.10.3</b> For multicategorical variables (&gt;2 subgroups)</a></li>
</ul></li>
<li class="chapter" data-level="6.11" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#reference"><i class="fa fa-check"></i><b>6.11</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html"><i class="fa fa-check"></i><b>7</b> <strong>TME Interaction analysis</strong></a>
<ul>
<li class="chapter" data-level="7.1" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#loading-packages-4"><i class="fa fa-check"></i><b>7.1</b> Loading packages</a></li>
<li class="chapter" data-level="7.2" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#downloading-data-for-example-3"><i class="fa fa-check"></i><b>7.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="7.3" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#gene-annotation-hgu133plus-2-affaymetrix"><i class="fa fa-check"></i><b>7.3</b> Gene Annotation: HGU133PLUS-2 (Affaymetrix)</a></li>
<li class="chapter" data-level="7.4" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#tme-deconvolution-using-cibersort-algorithm"><i class="fa fa-check"></i><b>7.4</b> TME deconvolution using CIBERSORT algorithm</a></li>
<li class="chapter" data-level="7.5" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#identifying-tme-patterns"><i class="fa fa-check"></i><b>7.5</b> Identifying TME patterns</a></li>
<li class="chapter" data-level="7.6" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#cell-abundance-of-each-cluster"><i class="fa fa-check"></i><b>7.6</b> Cell abundance of each cluster</a></li>
<li class="chapter" data-level="7.7" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#deg-analysis-between-tme-subtypes"><i class="fa fa-check"></i><b>7.7</b> DEG analysis between TME subtypes</a></li>
<li class="chapter" data-level="7.8" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#identifying-lr-associated-with-tme-clusters"><i class="fa fa-check"></i><b>7.8</b> Identifying LR associated with TME clusters</a></li>
<li class="chapter" data-level="7.9" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#identifying-signatures-associated-with-tme-clusters"><i class="fa fa-check"></i><b>7.9</b> Identifying signatures associated with TME clusters</a></li>
<li class="chapter" data-level="7.10" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#references-3"><i class="fa fa-check"></i><b>7.10</b> References</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html"><i class="fa fa-check"></i><b>8</b> <strong>Tumor ecosystem analysis</strong></a>
<ul>
<li class="chapter" data-level="8.1" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#loading-packages-5"><i class="fa fa-check"></i><b>8.1</b> Loading packages</a></li>
<li class="chapter" data-level="8.2" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#downloading-data-for-example-4"><i class="fa fa-check"></i><b>8.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="8.3" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#gene-annotation-hgu133plus-2-affaymetrix-1"><i class="fa fa-check"></i><b>8.3</b> Gene Annotation: HGU133PLUS-2 (Affaymetrix)</a></li>
<li class="chapter" data-level="8.4" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#determine-tme-subtype-of-gastric-cancer-using-tmeclassifier-r-package"><i class="fa fa-check"></i><b>8.4</b> Determine TME subtype of gastric cancer using TMEclassifier R package</a></li>
<li class="chapter" data-level="8.5" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#deg-analysis-method1"><i class="fa fa-check"></i><b>8.5</b> DEG analysis: method1</a></li>
<li class="chapter" data-level="8.6" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#gsea-analysis-based-on-differential-express-gene-analysis-results"><i class="fa fa-check"></i><b>8.6</b> GSEA analysis based on differential express gene analysis results</a></li>
<li class="chapter" data-level="8.7" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#deg-analysis-method2"><i class="fa fa-check"></i><b>8.7</b> DEG analysis: method2</a></li>
<li class="chapter" data-level="8.8" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#identifying-signatures-associated-with-tme-clusters-1"><i class="fa fa-check"></i><b>8.8</b> Identifying signatures associated with TME clusters</a></li>
<li class="chapter" data-level="8.9" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#references-4"><i class="fa fa-check"></i><b>8.9</b> References</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html"><i class="fa fa-check"></i><b>9</b> <strong>TME and genomic interaction</strong></a>
<ul>
<li class="chapter" data-level="9.1" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#loading-packages-6"><i class="fa fa-check"></i><b>9.1</b> Loading packages</a></li>
<li class="chapter" data-level="9.2" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#genomic-data-prepare"><i class="fa fa-check"></i><b>9.2</b> Genomic data prepare</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#using-tcgabiolinks-to-download-genomic-data"><i class="fa fa-check"></i><b>9.2.1</b> Using TCGAbiolinks to download genomic data</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#identifying-mutations-associated-with-tme"><i class="fa fa-check"></i><b>9.3</b> Identifying Mutations Associated with TME</a></li>
<li class="chapter" data-level="9.4" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#oncoprint-of-result"><i class="fa fa-check"></i><b>9.4</b> OncoPrint of result</a></li>
<li class="chapter" data-level="9.5" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#boxplot-of-top-10-mutated-genes"><i class="fa fa-check"></i><b>9.5</b> Boxplot of top 10 mutated genes</a></li>
<li class="chapter" data-level="9.6" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#other-methods-to-obtaind-genomic-data"><i class="fa fa-check"></i><b>9.6</b> Other methods to obtaind genomic data</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#using-tcgamutations"><i class="fa fa-check"></i><b>9.6.1</b> Using TCGAmutations</a></li>
<li class="chapter" data-level="9.6.2" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#using-maftools"><i class="fa fa-check"></i><b>9.6.2</b> Using maftools</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#references-5"><i class="fa fa-check"></i><b>9.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="tme-modeling.html"><a href="tme-modeling.html"><i class="fa fa-check"></i><b>10</b> <strong>TME Modeling</strong></a>
<ul>
<li class="chapter" data-level="10.1" data-path="tme-modeling.html"><a href="tme-modeling.html#loading-packages-7"><i class="fa fa-check"></i><b>10.1</b> Loading packages</a></li>
<li class="chapter" data-level="10.2" data-path="tme-modeling.html"><a href="tme-modeling.html#data-prepare"><i class="fa fa-check"></i><b>10.2</b> Data prepare</a></li>
<li class="chapter" data-level="10.3" data-path="tme-modeling.html"><a href="tme-modeling.html#input-data-overall-survival-prepare"><i class="fa fa-check"></i><b>10.3</b> Input data (overall survival) prepare</a></li>
<li class="chapter" data-level="10.4" data-path="tme-modeling.html"><a href="tme-modeling.html#constructing-survival-prediction-models"><i class="fa fa-check"></i><b>10.4</b> Constructing survival prediction models</a></li>
<li class="chapter" data-level="10.5" data-path="tme-modeling.html"><a href="tme-modeling.html#input-data-response-prepare"><i class="fa fa-check"></i><b>10.5</b> Input data (Response) prepare</a></li>
<li class="chapter" data-level="10.6" data-path="tme-modeling.html"><a href="tme-modeling.html#constructing-prediction-models-for-response"><i class="fa fa-check"></i><b>10.6</b> Constructing prediction models for response</a></li>
<li class="chapter" data-level="10.7" data-path="tme-modeling.html"><a href="tme-modeling.html#references-6"><i class="fa fa-check"></i><b>10.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html"><i class="fa fa-check"></i><b>11</b> <strong>Some tips about IOBR</strong></a>
<ul>
<li class="chapter" data-level="11.1" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html#loading-packages-8"><i class="fa fa-check"></i><b>11.1</b> Loading packages</a></li>
<li class="chapter" data-level="11.2" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html#colour-configuration"><i class="fa fa-check"></i><b>11.2</b> Colour Configuration</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html#categorical-variable"><i class="fa fa-check"></i><b>11.2.1</b> categorical variable</a></li>
<li class="chapter" data-level="11.2.2" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html#higher-number-of-subgroups"><i class="fa fa-check"></i><b>11.2.2</b> Higher number of subgroups</a></li>
<li class="chapter" data-level="11.2.3" data-path="some-tips-about-iobr.html"><a href="some-tips-about-iobr.html#gradient-colour-or-heatmap-colour-scheme"><i class="fa fa-check"></i><b>11.2.3</b> Gradient colour or heatmap colour scheme</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="references-7.html"><a href="references-7.html"><i class="fa fa-check"></i><b>12</b> <strong>References</strong></a>
<ul>
<li class="chapter" data-level="12.1" data-path="references-7.html"><a href="references-7.html#tme-deconvolution-1"><i class="fa fa-check"></i><b>12.1</b> TME deconvolution</a></li>
<li class="chapter" data-level="12.2" data-path="references-7.html"><a href="references-7.html#tme-signatures"><i class="fa fa-check"></i><b>12.2</b> TME Signatures</a></li>
<li class="chapter" data-level="12.3" data-path="references-7.html"><a href="references-7.html#data-sets"><i class="fa fa-check"></i><b>12.3</b> Data sets</a></li>
<li class="chapter" data-level="12.4" data-path="references-7.html"><a href="references-7.html#others"><i class="fa fa-check"></i><b>12.4</b> Others</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/DongqiangZeng0808" target="blank">Welcome to my GitHub</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">IOBR (Immuno-Oncology Biological Research)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="tumor-ecosystem-analysis" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">Chapter 8</span> <strong>Tumor ecosystem analysis</strong><a href="tumor-ecosystem-analysis.html#tumor-ecosystem-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="loading-packages-5" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Loading packages<a href="tumor-ecosystem-analysis.html#loading-packages-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="tumor-ecosystem-analysis.html#cb296-1" tabindex="-1"></a><span class="fu">library</span>(IOBR)</span></code></pre></div>
</div>
<div id="downloading-data-for-example-4" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> Downloading data for example<a href="tumor-ecosystem-analysis.html#downloading-data-for-example-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Obtaining data set from GEO <a href="https://pubmed.ncbi.nlm.nih.gov/25894828/">Gastric cancer: GSE62254</a> using <code>GEOquery</code> R package.</p>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb297-1"><a href="tumor-ecosystem-analysis.html#cb297-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;GEOquery&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;GEOquery&quot;</span>)</span>
<span id="cb297-2"><a href="tumor-ecosystem-analysis.html#cb297-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;GEOquery&quot;</span>)</span>
<span id="cb297-3"><a href="tumor-ecosystem-analysis.html#cb297-3" tabindex="-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: This process may take a few minutes which depends on the internet connection speed. Please wait for its completion.</span></span>
<span id="cb297-4"><a href="tumor-ecosystem-analysis.html#cb297-4" tabindex="-1"></a>eset_geo<span class="ot">&lt;-</span><span class="fu">getGEO</span>(<span class="at">GEO     =</span> <span class="st">&quot;GSE62254&quot;</span>, <span class="at">getGPL  =</span> F, <span class="at">destdir =</span> <span class="st">&quot;./&quot;</span>)</span>
<span id="cb297-5"><a href="tumor-ecosystem-analysis.html#cb297-5" tabindex="-1"></a>eset    <span class="ot">&lt;-</span>eset_geo[[<span class="dv">1</span>]]</span>
<span id="cb297-6"><a href="tumor-ecosystem-analysis.html#cb297-6" tabindex="-1"></a>eset    <span class="ot">&lt;-</span><span class="fu">exprs</span>(eset)</span>
<span id="cb297-7"><a href="tumor-ecosystem-analysis.html#cb297-7" tabindex="-1"></a>eset[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##           GSM1523727 GSM1523728 GSM1523729 GSM1523744 GSM1523745
## 1007_s_at  3.2176645  3.0624323  3.0279131   2.921683  2.8456013
## 1053_at    2.4050109  2.4394879  2.2442708   2.345916  2.4328582
## 117_at     1.4933412  1.8067380  1.5959665   1.839822  1.8326058
## 121_at     2.1965561  2.2812181  2.1865556   2.258599  2.1874363
## 1255_g_at  0.8698382  0.9502466  0.8125414   1.012860  0.9441993</code></pre>
</div>
<div id="gene-annotation-hgu133plus-2-affaymetrix-1" class="section level2 hasAnchor" number="8.3">
<h2><span class="header-section-number">8.3</span> Gene Annotation: HGU133PLUS-2 (Affaymetrix)<a href="tumor-ecosystem-analysis.html#gene-annotation-hgu133plus-2-affaymetrix-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="tumor-ecosystem-analysis.html#cb299-1" tabindex="-1"></a><span class="co"># Conduct gene annotation using `anno_hug133plus2` file; If identical gene symbols exists, these genes would be ordered by the mean expression levels. The gene symbol with highest mean expression level is selected and remove others. </span></span>
<span id="cb299-2"><a href="tumor-ecosystem-analysis.html#cb299-2" tabindex="-1"></a></span>
<span id="cb299-3"><a href="tumor-ecosystem-analysis.html#cb299-3" tabindex="-1"></a>eset<span class="ot">&lt;-</span><span class="fu">anno_eset</span>(<span class="at">eset       =</span> eset,</span>
<span id="cb299-4"><a href="tumor-ecosystem-analysis.html#cb299-4" tabindex="-1"></a>                <span class="at">annotation =</span> anno_hug133plus2,</span>
<span id="cb299-5"><a href="tumor-ecosystem-analysis.html#cb299-5" tabindex="-1"></a>                <span class="at">symbol     =</span> <span class="st">&quot;symbol&quot;</span>,</span>
<span id="cb299-6"><a href="tumor-ecosystem-analysis.html#cb299-6" tabindex="-1"></a>                <span class="at">probe      =</span> <span class="st">&quot;probe_id&quot;</span>,</span>
<span id="cb299-7"><a href="tumor-ecosystem-analysis.html#cb299-7" tabindex="-1"></a>                <span class="at">method     =</span> <span class="st">&quot;mean&quot;</span>)</span>
<span id="cb299-8"><a href="tumor-ecosystem-analysis.html#cb299-8" tabindex="-1"></a>eset[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>##              GSM1523727 GSM1523728 GSM1523729
## SH3KBP1        4.327974   4.316195   4.351425
## RPL41          4.246149   4.246808   4.257940
## EEF1A1         4.293762   4.291038   4.262199
## COX2           4.250288   4.283714   4.270508
## LOC101928826   4.219303   4.219670   4.213252</code></pre>
</div>
<div id="determine-tme-subtype-of-gastric-cancer-using-tmeclassifier-r-package" class="section level2 hasAnchor" number="8.4">
<h2><span class="header-section-number">8.4</span> Determine TME subtype of gastric cancer using <a href="https://github.com/LiaoWJLab/TMEclassifier">TMEclassifier R package</a><a href="tumor-ecosystem-analysis.html#determine-tme-subtype-of-gastric-cancer-using-tmeclassifier-r-package" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb301-1"><a href="tumor-ecosystem-analysis.html#cb301-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;TMEclassifier&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))  devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;LiaoWJLab/TMEclassifier&quot;</span>)</span>
<span id="cb301-2"><a href="tumor-ecosystem-analysis.html#cb301-2" tabindex="-1"></a><span class="fu">library</span>(TMEclassifier)</span>
<span id="cb301-3"><a href="tumor-ecosystem-analysis.html#cb301-3" tabindex="-1"></a>tme <span class="ot">&lt;-</span> <span class="fu">tme_classifier</span>(<span class="at">eset =</span> eset, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Step-1: Expression data preprocessing...</code></pre>
<pre><code>## Step-2: TME deconvolution...</code></pre>
<pre><code>## Step-3: Predicting TME phenotypes...</code></pre>
<pre><code>## [09:52:29] WARNING: src/learner.cc:1203: 
##   If you are loading a serialized model (like pickle in Python, RDS in R) generated by
##   older XGBoost, please export the model by calling `Booster.save_model` from that version
##   first, then load it back in current version. See:
## 
##     https://xgboost.readthedocs.io/en/latest/tutorials/saving_model.html
## 
##   for more details about differences between saving model and serializing.
## 
## [09:52:29] WARNING: src/learner.cc:888: Found JSON model saved before XGBoost 1.6, please save the model using current version again. The support for old JSON model will be discontinued in XGBoost 2.3.
## [09:52:29] WARNING: src/learner.cc:553: 
##   If you are loading a serialized model (like pickle in Python, RDS in R) generated by
##   older XGBoost, please export the model by calling `Booster.save_model` from that version
##   first, then load it back in current version. See:
## 
##     https://xgboost.readthedocs.io/en/latest/tutorials/saving_model.html
## 
##   for more details about differences between saving model and serializing.</code></pre>
<pre><code>## &gt;&gt;&gt;--- DONE!</code></pre>
<div class="sourceCode" id="cb307"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb307-1"><a href="tumor-ecosystem-analysis.html#cb307-1" tabindex="-1"></a><span class="fu">table</span>(tme<span class="sc">$</span>TMEcluster)</span></code></pre></div>
<pre><code>## 
##  IA  IE  IS 
## 107  96  97</code></pre>
<div class="sourceCode" id="cb309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb309-1"><a href="tumor-ecosystem-analysis.html#cb309-1" tabindex="-1"></a><span class="fu">head</span>(tme)</span></code></pre></div>
<pre><code>##           ID          IE         IS         IA TMEcluster
## 1 GSM1523727 0.204623557 0.11212681 0.68324962         IA
## 2 GSM1523728 0.009599504 0.11179146 0.87860903         IA
## 3 GSM1523729 0.852615046 0.11369089 0.03369407         IE
## 4 GSM1523744 0.053842233 0.06994632 0.87621145         IA
## 5 GSM1523745 0.055973019 0.80839488 0.13563209         IS
## 6 GSM1523746 0.545343299 0.37437568 0.08028102         IE</code></pre>
<div class="sourceCode" id="cb311"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb311-1"><a href="tumor-ecosystem-analysis.html#cb311-1" tabindex="-1"></a><span class="fu">table</span>(tme<span class="sc">$</span>TMEcluster)</span></code></pre></div>
<pre><code>## 
##  IA  IE  IS 
## 107  96  97</code></pre>
<div class="sourceCode" id="cb313"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb313-1"><a href="tumor-ecosystem-analysis.html#cb313-1" tabindex="-1"></a><span class="fu">head</span>(tme)</span></code></pre></div>
<pre><code>##           ID          IE         IS         IA TMEcluster
## 1 GSM1523727 0.204623557 0.11212681 0.68324962         IA
## 2 GSM1523728 0.009599504 0.11179146 0.87860903         IA
## 3 GSM1523729 0.852615046 0.11369089 0.03369407         IE
## 4 GSM1523744 0.053842233 0.06994632 0.87621145         IA
## 5 GSM1523745 0.055973019 0.80839488 0.13563209         IS
## 6 GSM1523746 0.545343299 0.37437568 0.08028102         IE</code></pre>
</div>
<div id="deg-analysis-method1" class="section level2 hasAnchor" number="8.5">
<h2><span class="header-section-number">8.5</span> DEG analysis: method1<a href="tumor-ecosystem-analysis.html#deg-analysis-method1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Differential analysis of selected immune-activated and immune-expelled gastric cancers</p>
<div class="sourceCode" id="cb315"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb315-1"><a href="tumor-ecosystem-analysis.html#cb315-1" tabindex="-1"></a>pdata <span class="ot">&lt;-</span> tme[<span class="sc">!</span>tme<span class="sc">$</span>TMEcluster<span class="sc">==</span><span class="st">&quot;IS&quot;</span>, ]</span>
<span id="cb315-2"><a href="tumor-ecosystem-analysis.html#cb315-2" tabindex="-1"></a>deg  <span class="ot">&lt;-</span>   <span class="fu">iobr_deg</span>(<span class="at">eset         =</span> eset,</span>
<span id="cb315-3"><a href="tumor-ecosystem-analysis.html#cb315-3" tabindex="-1"></a>                   <span class="at">annotation    =</span> <span class="cn">NULL</span>,</span>
<span id="cb315-4"><a href="tumor-ecosystem-analysis.html#cb315-4" tabindex="-1"></a>                   <span class="at">pdata        =</span> pdata,</span>
<span id="cb315-5"><a href="tumor-ecosystem-analysis.html#cb315-5" tabindex="-1"></a>                   <span class="at">group_id     =</span> <span class="st">&quot;TMEcluster&quot;</span>,</span>
<span id="cb315-6"><a href="tumor-ecosystem-analysis.html#cb315-6" tabindex="-1"></a>                   <span class="at">pdata_id     =</span> <span class="st">&quot;ID&quot;</span>,</span>
<span id="cb315-7"><a href="tumor-ecosystem-analysis.html#cb315-7" tabindex="-1"></a>                   <span class="at">array        =</span> <span class="cn">TRUE</span>,</span>
<span id="cb315-8"><a href="tumor-ecosystem-analysis.html#cb315-8" tabindex="-1"></a>                   <span class="at">method       =</span> <span class="st">&quot;limma&quot;</span>,</span>
<span id="cb315-9"><a href="tumor-ecosystem-analysis.html#cb315-9" tabindex="-1"></a>                   <span class="at">contrast     =</span> <span class="fu">c</span>(<span class="st">&quot;IA&quot;</span>,<span class="st">&quot;IE&quot;</span>),</span>
<span id="cb315-10"><a href="tumor-ecosystem-analysis.html#cb315-10" tabindex="-1"></a>                   <span class="at">path         =</span> <span class="st">&quot;result&quot;</span>,</span>
<span id="cb315-11"><a href="tumor-ecosystem-analysis.html#cb315-11" tabindex="-1"></a>                   <span class="at">padj_cutoff  =</span> <span class="fl">0.01</span>,</span>
<span id="cb315-12"><a href="tumor-ecosystem-analysis.html#cb315-12" tabindex="-1"></a>                   <span class="at">logfc_cutoff =</span> <span class="fl">0.5</span>)</span></code></pre></div>
<pre><code>## &gt;&gt;&gt;== Matching grouping information and expression matrix</code></pre>
<pre><code>## &gt;&gt;&gt;== limma was selected for differential gene analysis of Array data</code></pre>
<pre><code>## group1 = IE</code></pre>
<pre><code>## group2 = NA</code></pre>
<pre><code>## # A tibble: 6 √ó 11
##   symbol  log2FoldChange AveExpr     t   pvalue     padj     B sigORnot    label
##   &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt;
## 1 TMEM100          0.774    1.84  13.9 2.47e-31 5.37e-27  60.4 Up_regulat‚Ä¶ Both 
## 2 ABCA8            0.933    1.90  12.9 3.11e-28 3.38e-24  53.4 Up_regulat‚Ä¶ Both 
## 3 HHIP             0.613    1.73  12.1 7.62e-26 4.46e-22  48.0 Up_regulat‚Ä¶ Both 
## 4 LMNB2           -0.287    2.25 -12.1 9.28e-26 4.46e-22  47.8 NOT         Sign‚Ä¶
## 5 MCM6            -0.211    3.02 -12.1 1.02e-25 4.46e-22  47.7 NOT         Sign‚Ä¶
## 6 ADH1B            0.907    1.86  12.0 2.27e-25 7.04e-22  47.0 Up_regulat‚Ä¶ Both 
## # ‚Ñπ 2 more variables: IE &lt;dbl&gt;, `` &lt;dbl&gt;</code></pre>
</div>
<div id="gsea-analysis-based-on-differential-express-gene-analysis-results" class="section level2 hasAnchor" number="8.6">
<h2><span class="header-section-number">8.6</span> GSEA analysis based on differential express gene analysis results<a href="tumor-ecosystem-analysis.html#gsea-analysis-based-on-differential-express-gene-analysis-results" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Select the gene set list in IOBR‚Äôs signature collection.</p>
<div class="sourceCode" id="cb321"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb321-1"><a href="tumor-ecosystem-analysis.html#cb321-1" tabindex="-1"></a><span class="fu">head</span>(deg)</span></code></pre></div>
<pre><code>## # A tibble: 6 √ó 11
##   symbol  log2FoldChange AveExpr     t   pvalue     padj     B sigORnot    label
##   &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt;
## 1 TMEM100          0.774    1.84  13.9 2.47e-31 5.37e-27  60.4 Up_regulat‚Ä¶ Both 
## 2 ABCA8            0.933    1.90  12.9 3.11e-28 3.38e-24  53.4 Up_regulat‚Ä¶ Both 
## 3 HHIP             0.613    1.73  12.1 7.62e-26 4.46e-22  48.0 Up_regulat‚Ä¶ Both 
## 4 LMNB2           -0.287    2.25 -12.1 9.28e-26 4.46e-22  47.8 NOT         Sign‚Ä¶
## 5 MCM6            -0.211    3.02 -12.1 1.02e-25 4.46e-22  47.7 NOT         Sign‚Ä¶
## 6 ADH1B            0.907    1.86  12.0 2.27e-25 7.04e-22  47.0 Up_regulat‚Ä¶ Both 
## # ‚Ñπ 2 more variables: IE &lt;dbl&gt;, `` &lt;dbl&gt;</code></pre>
<div class="sourceCode" id="cb323"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb323-1"><a href="tumor-ecosystem-analysis.html#cb323-1" tabindex="-1"></a>sig_list <span class="ot">&lt;-</span> signature_collection[<span class="fu">c</span>(<span class="st">&quot;TMEscoreB_CIR&quot;</span>, <span class="st">&quot;TMEscoreA_CIR&quot;</span>, <span class="st">&quot;DNA_replication&quot;</span>, <span class="st">&quot;Base_excision_repair&quot;</span>,</span>
<span id="cb323-2"><a href="tumor-ecosystem-analysis.html#cb323-2" tabindex="-1"></a>                                   <span class="st">&quot;Pan_F_TBRs&quot;</span>, <span class="st">&quot;TGFb.myCAF&quot;</span>, <span class="st">&quot;Ferroptosis&quot;</span>, <span class="st">&quot;TLS_Nature&quot;</span>, <span class="st">&quot;Glycolysis&quot;</span>)]</span>
<span id="cb323-3"><a href="tumor-ecosystem-analysis.html#cb323-3" tabindex="-1"></a>sig_list</span></code></pre></div>
<pre><code>## $TMEscoreB_CIR
##   [1] &quot;DCN&quot;          &quot;SEPP1&quot;        &quot;ACTA2&quot;        &quot;SPARCL1&quot;      &quot;BEX3&quot;        
##   [6] &quot;MYLK&quot;         &quot;AKR1C1&quot;       &quot;TIMP2&quot;        &quot;MXRA7&quot;        &quot;C11orf96&quot;    
##  [11] &quot;CAV1&quot;         &quot;PDGFRA&quot;       &quot;FHL1&quot;         &quot;MGP&quot;          &quot;EID1&quot;        
##  [16] &quot;LOC101930400&quot; &quot;DST&quot;          &quot;GREM1&quot;        &quot;FERMT2&quot;       &quot;TNC&quot;         
##  [21] &quot;CYBRD1&quot;       &quot;LTBP1&quot;        &quot;ACTG2&quot;        &quot;TMEM47&quot;       &quot;SERPINE2&quot;    
##  [26] &quot;ANTXR2&quot;       &quot;GNG11&quot;        &quot;TAGLN&quot;        &quot;GSTA4&quot;        &quot;PKIG&quot;        
##  [31] &quot;MAOA&quot;         &quot;PTRF&quot;         &quot;FAM3B&quot;        &quot;PBX1&quot;         &quot;WLS&quot;         
##  [36] &quot;SELM&quot;         &quot;SVIL&quot;         &quot;MYH11&quot;        &quot;AGT&quot;          &quot;SPON1&quot;       
##  [41] &quot;TGFB1I1&quot;      &quot;PDLIM3&quot;       &quot;PDK4&quot;         &quot;SYNPO2&quot;       &quot;MSRB3&quot;       
##  [46] &quot;PROS1&quot;        &quot;EDNRA&quot;        &quot;AKAP12&quot;       &quot;PSD3&quot;         &quot;TNS1&quot;        
##  [51] &quot;JAM3&quot;         &quot;PDZRN3&quot;       &quot;DDR2&quot;         &quot;HMGCS2&quot;       &quot;SGCE&quot;        
##  [56] &quot;MRVI1&quot;        &quot;WFDC1&quot;        &quot;FBLN1&quot;        &quot;FMO5&quot;         &quot;MAOB&quot;        
##  [61] &quot;AMOTL1&quot;       &quot;AKT3&quot;         &quot;CNRIP1&quot;       &quot;CPE&quot;          &quot;MAP1B&quot;       
##  [66] &quot;RBP1&quot;         &quot;GNAI1&quot;        &quot;FOXF2&quot;        &quot;SORBS2&quot;       &quot;ZCCHC24&quot;     
##  [71] &quot;ZNF704&quot;       &quot;ARMCX1&quot;       &quot;DIXDC1&quot;       &quot;SSTR1&quot;        &quot;THRB&quot;        
##  [76] &quot;C3orf70&quot;      &quot;PKIB&quot;         &quot;CNN1&quot;         &quot;SYTL5&quot;        &quot;DACT1&quot;       
##  [81] &quot;SYNPO&quot;        &quot;GAS1&quot;         &quot;DPYSL3&quot;       &quot;CCDC80&quot;       &quot;TSPYL5&quot;      
##  [86] &quot;DCHS1&quot;        &quot;SOBP&quot;         &quot;AOC3&quot;         &quot;NDN&quot;          &quot;FGF7P3&quot;      
##  [91] &quot;SMAD9&quot;        &quot;MCC&quot;          &quot;CLMP&quot;         &quot;MYL9&quot;         &quot;RBP4&quot;        
##  [96] &quot;PLN&quot;          &quot;SPOCK1&quot;       &quot;COL14A1&quot;      &quot;CRYAB&quot;        &quot;SRPX&quot;        
## [101] &quot;EML1&quot;         &quot;RERG&quot;         &quot;PPP1R3C&quot;      &quot;LOC100506718&quot; &quot;CH25H&quot;       
## [106] &quot;HSPB8&quot;        &quot;PID1&quot;         &quot;TTC28&quot;        &quot;STON1&quot;        &quot;ABCG2&quot;       
## [111] &quot;ZSCAN18&quot;      &quot;SCIN&quot;         &quot;C14orf132&quot;    &quot;TMEM55A&quot;      &quot;WASF3&quot;       
## [116] &quot;PAPLN&quot;        &quot;COLEC12&quot;      &quot;ACKR1&quot;        &quot;TMEM150C&quot;     &quot;RAI2&quot;        
## [121] &quot;TSPAN7&quot;       &quot;MRGPRF&quot;       &quot;ABCA8&quot;        &quot;CHIC1&quot;        &quot;NBEA&quot;        
## [126] &quot;FAM13C&quot;       &quot;SETBP1&quot;       &quot;LDOC1&quot;        &quot;TMEM100&quot;      &quot;LOC101930349&quot;
## [131] &quot;PRICKLE2&quot;     &quot;TSPAN18&quot;      &quot;FABP4&quot;        &quot;ARHGEF26&quot;     &quot;ERICH5&quot;      
## [136] &quot;MYOCD&quot;        &quot;BEX2&quot;         &quot;PPP1R14A&quot;     &quot;FGF13&quot;        &quot;RUNX1T1&quot;     
## [141] &quot;MAGI2-AS3&quot;    &quot;LINC01279&quot;    &quot;REEP1&quot;        &quot;PLAC9&quot;        &quot;MYEF2&quot;       
## [146] &quot;PRKD1&quot;        &quot;RGN&quot;          &quot;CLDN11&quot;       &quot;ANK2&quot;         &quot;ESRRG&quot;       
## [151] &quot;SYNC&quot;         &quot;ZNF667-AS1&quot;   &quot;FGF7&quot;         &quot;SFRP1&quot;        &quot;HMCN1&quot;       
## [156] &quot;TCEAL7&quot;       &quot;OGN&quot;          &quot;MAGI2&quot;        &quot;MIR100HG&quot;     &quot;FILIP1&quot;      
## [161] &quot;LOC100507334&quot; &quot;ANKRD6&quot;       &quot;PLEKHH2&quot;      &quot;ZNF542P&quot;      &quot;ARMCX4&quot;      
## [166] &quot;NOV&quot;          &quot;DCLK1&quot;        &quot;ARHGAP28&quot;     &quot;C2orf40&quot;      &quot;TRHDE&quot;       
## [171] &quot;EPHA7&quot;        &quot;SCRG1&quot;        &quot;ZNF677&quot;       &quot;ZFPM2&quot;        &quot;PEG3&quot;        
## [176] &quot;SERP2&quot;        &quot;ZNF415&quot;       &quot;MAMDC2&quot;       &quot;RBM24&quot;        &quot;MEOX2&quot;       
## 
## $TMEscoreA_CIR
##  [1] &quot;HLA-DPB1&quot;       &quot;UBD&quot;            &quot;LOC100509457&quot;   &quot;WARS&quot;          
##  [5] &quot;TAP1&quot;           &quot;HLA-DMA&quot;        &quot;TRIM22&quot;         &quot;PSAT1&quot;         
##  [9] &quot;CXCL10&quot;         &quot;SOCS3&quot;          &quot;CXCL9&quot;          &quot;PBK&quot;           
## [13] &quot;CCL4&quot;           &quot;CCL5&quot;           &quot;BCL2A1&quot;         &quot;TRBC1&quot;         
## [17] &quot;IDO1&quot;           &quot;NFE2L3&quot;         &quot;CCL3L3&quot;         &quot;DTL&quot;           
## [21] &quot;MMP9&quot;           &quot;SLC2A3&quot;         &quot;ZNF367&quot;         &quot;RCC1&quot;          
## [25] &quot;STIL&quot;           &quot;TRAC&quot;           &quot;HELLS&quot;          &quot;GZMB&quot;          
## [29] &quot;RTEL1-TNFRSF6B&quot; &quot;CXCL11&quot;         &quot;GBP5&quot;           &quot;CD2&quot;           
## [33] &quot;CDCA2&quot;          &quot;CDT1&quot;           &quot;TNFAIP2&quot;        &quot;TYMP&quot;          
## [37] &quot;MICB&quot;           &quot;SLC2A14&quot;        &quot;GZMK&quot;           &quot;CD8A&quot;          
## [41] &quot;CENPH&quot;          &quot;MND1&quot;           &quot;BATF2&quot;          &quot;BRIP1&quot;         
## [45] &quot;E2F7&quot;           &quot;KIF18A&quot;         &quot;AIM2&quot;           &quot;ETV7&quot;          
## [49] &quot;ITK&quot;            &quot;GNLY&quot;           &quot;GPR171&quot;         &quot;WDHD1&quot;         
## [53] &quot;GBP4&quot;           &quot;MB21D1&quot;         &quot;NLRP3&quot;          &quot;MCEMP1&quot;        
## [57] &quot;POLR3G&quot;         &quot;NLRC3&quot;          &quot;KLRC2&quot;          &quot;CLEC5A&quot;        
## [61] &quot;ARHGAP11A&quot;      &quot;GPR84&quot;          &quot;IFNG&quot;           &quot;ZBED2&quot;         
## 
## $DNA_replication
##  [1] &quot;RNASEH2A&quot; &quot;POLD3&quot;    &quot;DNA2&quot;     &quot;FEN1&quot;     &quot;POLA2&quot;    &quot;RNASEH1&quot; 
##  [7] &quot;RPA4&quot;     &quot;LIG1&quot;     &quot;MCM2&quot;     &quot;MCM3&quot;     &quot;MCM4&quot;     &quot;MCM5&quot;    
## [13] &quot;MCM6&quot;     &quot;MCM7&quot;     &quot;PCNA&quot;     &quot;POLE3&quot;    &quot;POLA1&quot;    &quot;POLD1&quot;   
## [19] &quot;POLD2&quot;    &quot;POLE&quot;     &quot;POLE2&quot;    &quot;PRIM1&quot;    &quot;PRIM2&quot;    &quot;POLE4&quot;   
## [25] &quot;POLD4&quot;    &quot;RFC1&quot;     &quot;RFC2&quot;     &quot;RFC3&quot;     &quot;RFC4&quot;     &quot;RFC5&quot;    
## [31] &quot;RPA1&quot;     &quot;RPA2&quot;     &quot;RPA3&quot;     &quot;SSBP1&quot;    &quot;RNASEH2B&quot; &quot;RNASEH2C&quot;
## 
## $Base_excision_repair
##  [1] &quot;PARP2&quot; &quot;PARP3&quot; &quot;POLD3&quot; &quot;PARP1&quot; &quot;PARP4&quot; &quot;FEN1&quot;  &quot;SMUG1&quot; &quot;NEIL2&quot; &quot;APEX2&quot;
## [10] &quot;POLL&quot;  &quot;HMGB1&quot; &quot;APEX1&quot; &quot;LIG1&quot;  &quot;LIG3&quot;  &quot;MPG&quot;   &quot;MUTYH&quot; &quot;NTHL1&quot; &quot;OGG1&quot; 
## [19] &quot;PCNA&quot;  &quot;POLE3&quot; &quot;POLB&quot;  &quot;POLD1&quot; &quot;POLD2&quot; &quot;POLE&quot;  &quot;POLE2&quot; &quot;NEIL3&quot; &quot;POLE4&quot;
## [28] &quot;POLD4&quot; &quot;UNG&quot;   &quot;XRCC1&quot; &quot;NEIL1&quot; &quot;MBD4&quot; 
## 
## $Pan_F_TBRs
##  [1] &quot;ACTA2&quot;    &quot;ACTG2&quot;    &quot;ADAM12&quot;   &quot;ADAM19&quot;   &quot;CNN1&quot;     &quot;COL4A1&quot;  
##  [7] &quot;CTGF&quot;     &quot;CTPS1&quot;    &quot;FAM101B&quot;  &quot;FSTL3&quot;    &quot;HSPB1&quot;    &quot;IGFBP3&quot;  
## [13] &quot;PXDC1&quot;    &quot;SEMA7A&quot;   &quot;SH3PXD2A&quot; &quot;TAGLN&quot;    &quot;TGFBI&quot;    &quot;TNS1&quot;    
## [19] &quot;TPM1&quot;    
## 
## $&lt;NA&gt;
## NULL
## 
## $Ferroptosis
##  [1] &quot;ACSL4&quot;      &quot;AKR1C1-3&quot;   &quot;ALOXs&quot;      &quot;ATP5G3&quot;     &quot;CARS&quot;      
##  [6] &quot;CBS&quot;        &quot;CD44v&quot;      &quot;CHAC1&quot;      &quot;CISD1&quot;      &quot;CS&quot;        
## [11] &quot;DPP4&quot;       &quot;FANCD2&quot;     &quot;GCLC/GCLM&quot;  &quot;GLS2&quot;       &quot;GPX4&quot;      
## [16] &quot;GSS&quot;        &quot;HMGCR&quot;      &quot;HSPB1/5&quot;    &quot;KOD&quot;        &quot;LPCAT3&quot;    
## [21] &quot;MT1G&quot;       &quot;NCOA4&quot;      &quot;NFE2L2&quot;     &quot;PTGS2&quot;      &quot;RPL8&quot;      
## [26] &quot;SAT1&quot;       &quot;SLC7A11&quot;    &quot;SQS&quot;        &quot;TFRC&quot;       &quot;TP53&quot;      
## [31] &quot;TTC35/EMC2&quot; &quot;MESH1&quot;     
## 
## $TLS_Nature
## [1] &quot;CD79B&quot;  &quot;CD1D&quot;   &quot;CCR6&quot;   &quot;LAT&quot;    &quot;SKAP1&quot;  &quot;CETP&quot;   &quot;EIF1AY&quot; &quot;RBP5&quot;  
## [9] &quot;PTGDS&quot; 
## 
## $Glycolysis
##  [1] &quot;ACSS1&quot;   &quot;ACSS2&quot;   &quot;ADH1A&quot;   &quot;ADH1B&quot;   &quot;ADH1C&quot;   &quot;ADH4&quot;    &quot;ADH5&quot;   
##  [8] &quot;ADH6&quot;    &quot;ADH7&quot;    &quot;ADPGK&quot;   &quot;AKR1A1&quot;  &quot;ALDH1A3&quot; &quot;ALDH1B1&quot; &quot;ALDH2&quot;  
## [15] &quot;ALDH3A1&quot; &quot;ALDH3A2&quot; &quot;ALDH3B1&quot; &quot;ALDH3B2&quot; &quot;ALDH7A1&quot; &quot;ALDH9A1&quot; &quot;ALDOA&quot;  
## [22] &quot;ALDOB&quot;   &quot;ALDOC&quot;   &quot;BPGM&quot;    &quot;DLAT&quot;    &quot;DLD&quot;     &quot;ENO1&quot;    &quot;ENO2&quot;   
## [29] &quot;ENO3&quot;    &quot;FBP1&quot;    &quot;FBP2&quot;    &quot;G6PC&quot;    &quot;G6PC2&quot;   &quot;GALM&quot;    &quot;GAPDH&quot;  
## [36] &quot;GAPDHS&quot;  &quot;GCK&quot;     &quot;GPI&quot;     &quot;HK1&quot;     &quot;HK2&quot;     &quot;HK3&quot;     &quot;HKDC1&quot;  
## [43] &quot;LDHA&quot;    &quot;LDHAL6A&quot; &quot;LDHAL6B&quot; &quot;LDHB&quot;    &quot;LDHC&quot;    &quot;PANK1&quot;   &quot;PCK1&quot;   
## [50] &quot;PCK2&quot;    &quot;PDHA1&quot;   &quot;PDHA2&quot;   &quot;PDHB&quot;    &quot;PFKFB1&quot;  &quot;PFKFB2&quot;  &quot;PFKFB3&quot; 
## [57] &quot;PFKFB4&quot;  &quot;PFKL&quot;    &quot;PFKM&quot;    &quot;PFKP&quot;    &quot;PGAM1&quot;   &quot;PGAM2&quot;   &quot;PGAM4&quot;  
## [64] &quot;PGK1&quot;    &quot;PGK2&quot;    &quot;PGM1&quot;    &quot;PGM2&quot;    &quot;PKLR&quot;    &quot;PKM&quot;     &quot;SLC2A2&quot; 
## [71] &quot;TPI1&quot;</code></pre>
<div class="sourceCode" id="cb325"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb325-1"><a href="tumor-ecosystem-analysis.html#cb325-1" tabindex="-1"></a>gsea<span class="ot">&lt;-</span>     <span class="fu">sig_gsea</span>(deg,</span>
<span id="cb325-2"><a href="tumor-ecosystem-analysis.html#cb325-2" tabindex="-1"></a>                    <span class="at">genesets          =</span> sig_list,</span>
<span id="cb325-3"><a href="tumor-ecosystem-analysis.html#cb325-3" tabindex="-1"></a>                    <span class="at">path              =</span> <span class="st">&quot;result&quot;</span>,</span>
<span id="cb325-4"><a href="tumor-ecosystem-analysis.html#cb325-4" tabindex="-1"></a>                    <span class="at">gene_symbol       =</span> <span class="st">&quot;symbol&quot;</span>,</span>
<span id="cb325-5"><a href="tumor-ecosystem-analysis.html#cb325-5" tabindex="-1"></a>                    <span class="at">logfc             =</span> <span class="st">&quot;log2FoldChange&quot;</span>,</span>
<span id="cb325-6"><a href="tumor-ecosystem-analysis.html#cb325-6" tabindex="-1"></a>                    <span class="at">org               =</span> <span class="st">&quot;hsa&quot;</span>,</span>
<span id="cb325-7"><a href="tumor-ecosystem-analysis.html#cb325-7" tabindex="-1"></a>                    <span class="at">show_plot         =</span> <span class="cn">FALSE</span>,</span>
<span id="cb325-8"><a href="tumor-ecosystem-analysis.html#cb325-8" tabindex="-1"></a>                    <span class="at">msigdb            =</span> <span class="cn">TRUE</span>,</span>
<span id="cb325-9"><a href="tumor-ecosystem-analysis.html#cb325-9" tabindex="-1"></a>                    <span class="at">category          =</span> <span class="st">&quot;H&quot;</span>,</span>
<span id="cb325-10"><a href="tumor-ecosystem-analysis.html#cb325-10" tabindex="-1"></a>                    <span class="at">subcategory       =</span> <span class="cn">NULL</span>,</span>
<span id="cb325-11"><a href="tumor-ecosystem-analysis.html#cb325-11" tabindex="-1"></a>                    <span class="at">palette_bar       =</span> <span class="st">&quot;set2&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-9"></span>
<img src="fig/gsea-1.png" alt="GSEA of TME gent sets" width="95%" />
<p class="caption">
Figure 8.1: GSEA of TME gent sets
</p>
</div>
<p>Hallmark gene signatures</p>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb326-1"><a href="tumor-ecosystem-analysis.html#cb326-1" tabindex="-1"></a>gsea<span class="ot">&lt;-</span>     <span class="fu">sig_gsea</span>(deg,</span>
<span id="cb326-2"><a href="tumor-ecosystem-analysis.html#cb326-2" tabindex="-1"></a>                    <span class="at">genesets          =</span> <span class="cn">NULL</span>,</span>
<span id="cb326-3"><a href="tumor-ecosystem-analysis.html#cb326-3" tabindex="-1"></a>                    <span class="at">path              =</span> <span class="st">&quot;GSEA&quot;</span>,</span>
<span id="cb326-4"><a href="tumor-ecosystem-analysis.html#cb326-4" tabindex="-1"></a>                    <span class="at">gene_symbol       =</span> <span class="st">&quot;symbol&quot;</span>,</span>
<span id="cb326-5"><a href="tumor-ecosystem-analysis.html#cb326-5" tabindex="-1"></a>                    <span class="at">logfc             =</span> <span class="st">&quot;log2FoldChange&quot;</span>,</span>
<span id="cb326-6"><a href="tumor-ecosystem-analysis.html#cb326-6" tabindex="-1"></a>                    <span class="at">org               =</span> <span class="st">&quot;hsa&quot;</span>,</span>
<span id="cb326-7"><a href="tumor-ecosystem-analysis.html#cb326-7" tabindex="-1"></a>                    <span class="at">show_plot         =</span> <span class="cn">FALSE</span>,</span>
<span id="cb326-8"><a href="tumor-ecosystem-analysis.html#cb326-8" tabindex="-1"></a>                    <span class="at">msigdb            =</span> <span class="cn">TRUE</span>,</span>
<span id="cb326-9"><a href="tumor-ecosystem-analysis.html#cb326-9" tabindex="-1"></a>                    <span class="at">category          =</span> <span class="st">&quot;H&quot;</span>,</span>
<span id="cb326-10"><a href="tumor-ecosystem-analysis.html#cb326-10" tabindex="-1"></a>                    <span class="at">subcategory       =</span> <span class="cn">NULL</span>,</span>
<span id="cb326-11"><a href="tumor-ecosystem-analysis.html#cb326-11" tabindex="-1"></a>                    <span class="at">palette_bar       =</span> <span class="st">&quot;aaas&quot;</span>,</span>
<span id="cb326-12"><a href="tumor-ecosystem-analysis.html#cb326-12" tabindex="-1"></a>                    <span class="at">show_bar          =</span> <span class="dv">5</span>,</span>
<span id="cb326-13"><a href="tumor-ecosystem-analysis.html#cb326-13" tabindex="-1"></a>                    <span class="at">show_gsea         =</span> <span class="dv">6</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-11"></span>
<img src="fig/gsea-2.png" alt="GSEA of Hallmark gent sets" width="95%" />
<p class="caption">
Figure 8.2: GSEA of Hallmark gent sets
</p>
</div>
</div>
<div id="deg-analysis-method2" class="section level2 hasAnchor" number="8.7">
<h2><span class="header-section-number">8.7</span> DEG analysis: method2<a href="tumor-ecosystem-analysis.html#deg-analysis-method2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Identifing TME subtype-related differential genes using <code>find_markers_in_bulk</code></p>
<div class="sourceCode" id="cb327"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb327-1"><a href="tumor-ecosystem-analysis.html#cb327-1" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb327-2"><a href="tumor-ecosystem-analysis.html#cb327-2" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">find_markers_in_bulk</span>(<span class="at">pdata      =</span> tme, </span>
<span id="cb327-3"><a href="tumor-ecosystem-analysis.html#cb327-3" tabindex="-1"></a>                            <span class="at">eset       =</span> eset, </span>
<span id="cb327-4"><a href="tumor-ecosystem-analysis.html#cb327-4" tabindex="-1"></a>                            <span class="at">group      =</span> <span class="st">&quot;TMEcluster&quot;</span>, </span>
<span id="cb327-5"><a href="tumor-ecosystem-analysis.html#cb327-5" tabindex="-1"></a>                            <span class="at">nfeatures  =</span> <span class="dv">2000</span>, </span>
<span id="cb327-6"><a href="tumor-ecosystem-analysis.html#cb327-6" tabindex="-1"></a>                            <span class="at">top_n      =</span> <span class="dv">50</span>, </span>
<span id="cb327-7"><a href="tumor-ecosystem-analysis.html#cb327-7" tabindex="-1"></a>                            <span class="at">thresh.use =</span> <span class="fl">0.15</span>, </span>
<span id="cb327-8"><a href="tumor-ecosystem-analysis.html#cb327-8" tabindex="-1"></a>                            <span class="at">only.pos   =</span> <span class="cn">TRUE</span>, </span>
<span id="cb327-9"><a href="tumor-ecosystem-analysis.html#cb327-9" tabindex="-1"></a>                            <span class="at">min.pct    =</span> <span class="fl">0.10</span>)</span>
<span id="cb327-10"><a href="tumor-ecosystem-analysis.html#cb327-10" tabindex="-1"></a>top15 <span class="ot">&lt;-</span>  res<span class="sc">$</span>top_markers <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span> <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span>  dplyr<span class="sc">::</span><span class="fu">top_n</span>(<span class="dv">15</span>, avg_log2FC)</span>
<span id="cb327-11"><a href="tumor-ecosystem-analysis.html#cb327-11" tabindex="-1"></a>top15<span class="sc">$</span>gene</span></code></pre></div>
<pre><code>##  [1] &quot;IDO1&quot;           &quot;CXCL11&quot;         &quot;SLCO1B3&quot;        &quot;IFNG&quot;          
##  [5] &quot;AIM2&quot;           &quot;GZMB&quot;           &quot;VSNL1&quot;          &quot;CXCL10&quot;        
##  [9] &quot;CXCL9&quot;          &quot;GBP4&quot;           &quot;KLRC2&quot;          &quot;GNLY&quot;          
## [13] &quot;GZMH&quot;           &quot;KISS1R&quot;         &quot;WARS&quot;           &quot;C2orf40&quot;       
## [17] &quot;OGN&quot;            &quot;PGA4&quot;           &quot;SCN7A&quot;          &quot;C7&quot;            
## [21] &quot;ADH1B&quot;          &quot;GIF&quot;            &quot;SCRG1&quot;          &quot;LIPF&quot;          
## [25] &quot;GHRL&quot;           &quot;MAMDC2&quot;         &quot;VIP&quot;            &quot;ABCA8&quot;         
## [29] &quot;ATP4A&quot;          &quot;TMEM100&quot;        &quot;REG1B&quot;          &quot;MAGEA12&quot;       
## [33] &quot;MAGEA4&quot;         &quot;IL1A&quot;           &quot;PI15&quot;           &quot;IL11&quot;          
## [37] &quot;MAGEA6&quot;         &quot;MAGEA10-MAGEA5&quot; &quot;PPBP&quot;           &quot;PROK2&quot;         
## [41] &quot;MAGEA2B&quot;        &quot;CLEC5A&quot;         &quot;IL24&quot;           &quot;CTAG1A&quot;        
## [45] &quot;EREG&quot;</code></pre>
<p>Heatmap visualisation using <code>Seurat</code>‚Äôs <code>DoHeatmap</code></p>
<div class="sourceCode" id="cb329"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb329-1"><a href="tumor-ecosystem-analysis.html#cb329-1" tabindex="-1"></a><span class="co">#ÂÆö‰πâÂàÜÂûãÂØπÂ∫îÁöÑÈ¢úËâ≤</span></span>
<span id="cb329-2"><a href="tumor-ecosystem-analysis.html#cb329-2" tabindex="-1"></a>cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;#2692a4&#39;</span>,<span class="st">&#39;#fc0d3a&#39;</span>,<span class="st">&#39;#ffbe0b&#39;</span>)</span>
<span id="cb329-3"><a href="tumor-ecosystem-analysis.html#cb329-3" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">DoHeatmap</span>(res<span class="sc">$</span>sce, top15<span class="sc">$</span>gene, <span class="at">group.colors =</span> cols )<span class="sc">+</span></span>
<span id="cb329-4"><a href="tumor-ecosystem-analysis.html#cb329-4" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">colours =</span> <span class="fu">rev</span>(<span class="fu">colorRampPalette</span>(RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">11</span>,<span class="st">&quot;RdBu&quot;</span>))(<span class="dv">256</span>)))</span></code></pre></div>
<p>Extracting variables from the expression matrix to merge with TME subtypes</p>
<div class="sourceCode" id="cb330"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb330-1"><a href="tumor-ecosystem-analysis.html#cb330-1" tabindex="-1"></a>input <span class="ot">&lt;-</span> <span class="fu">combine_pd_eset</span>(<span class="at">eset =</span> eset, <span class="at">pdata =</span> tme, <span class="at">feas =</span> top15<span class="sc">$</span>gene, <span class="at">scale =</span> T)</span>
<span id="cb330-2"><a href="tumor-ecosystem-analysis.html#cb330-2" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">sig_box</span>(input, <span class="at">variable =</span> <span class="st">&quot;TMEcluster&quot;</span>, <span class="at">signature =</span> <span class="st">&quot;IFNG&quot;</span>, <span class="at">jitter =</span> <span class="cn">TRUE</span>,</span>
<span id="cb330-3"><a href="tumor-ecosystem-analysis.html#cb330-3" tabindex="-1"></a>              <span class="at">cols =</span>  cols, <span class="at">show_pvalue =</span> <span class="cn">TRUE</span>, <span class="at">size_of_pvalue =</span> <span class="dv">4</span>)</span></code></pre></div>
<pre><code>## # A tibble: 3 √ó 8
##   .y.       group1 group2        p    p.adj p.format p.signif method  
##   &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;     &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;   
## 1 signature IA     IE     4.09e-17 1.20e-16 &lt; 2e-16  ****     Wilcoxon
## 2 signature IA     IS     1.44e-13 2.90e-13 1.4e-13  ****     Wilcoxon
## 3 signature IE     IS     8.35e- 2 8.4 e- 2 0.084    ns       Wilcoxon</code></pre>
<div class="sourceCode" id="cb332"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb332-1"><a href="tumor-ecosystem-analysis.html#cb332-1" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">sig_box</span>(input, <span class="at">variable =</span> <span class="st">&quot;TMEcluster&quot;</span>, <span class="at">signature =</span> <span class="st">&quot;IL1A&quot;</span>, </span>
<span id="cb332-2"><a href="tumor-ecosystem-analysis.html#cb332-2" tabindex="-1"></a>              <span class="at">jitter =</span> <span class="cn">TRUE</span>, <span class="at">cols =</span>  cols, <span class="at">show_pvalue =</span> <span class="cn">TRUE</span>, <span class="at">size_of_pvalue =</span> <span class="dv">4</span>)</span></code></pre></div>
<pre><code>## # A tibble: 3 √ó 8
##   .y.       group1 group2        p    p.adj p.format p.signif method  
##   &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;     &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;   
## 1 signature IA     IE     1.46e-10 2.90e-10 1.5e-10  ****     Wilcoxon
## 2 signature IA     IS     8.22e- 7 8.2 e- 7 8.2e-07  ****     Wilcoxon
## 3 signature IE     IS     4.90e-20 1.5 e-19 &lt; 2e-16  ****     Wilcoxon</code></pre>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb334-1"><a href="tumor-ecosystem-analysis.html#cb334-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;patchwork&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))   <span class="fu">install.packages</span>(<span class="st">&quot;patchwork&quot;</span>)</span>
<span id="cb334-2"><a href="tumor-ecosystem-analysis.html#cb334-2" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb334-3"><a href="tumor-ecosystem-analysis.html#cb334-3" tabindex="-1"></a>p <span class="ot">&lt;-</span> (p1<span class="sc">|</span>p2<span class="sc">/</span>p3) <span class="sc">+</span> <span class="fu">plot_layout</span>(<span class="at">widths =</span> <span class="fu">c</span>(<span class="fl">2.3</span>,<span class="dv">1</span>))</span>
<span id="cb334-4"><a href="tumor-ecosystem-analysis.html#cb334-4" tabindex="-1"></a>p <span class="sc">+</span> <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">&#39;A&#39;</span>)</span></code></pre></div>
<p><img src="tme-deg-gsea_files/figure-html/unnamed-chunk-15-1.png" width="1152" /></p>
</div>
<div id="identifying-signatures-associated-with-tme-clusters-1" class="section level2 hasAnchor" number="8.8">
<h2><span class="header-section-number">8.8</span> Identifying signatures associated with TME clusters<a href="tumor-ecosystem-analysis.html#identifying-signatures-associated-with-tme-clusters-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Calculate TME associated signatures-(through PCA method).</p>
<div class="sourceCode" id="cb335"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb335-1"><a href="tumor-ecosystem-analysis.html#cb335-1" tabindex="-1"></a>sig_tme<span class="ot">&lt;-</span><span class="fu">calculate_sig_score</span>(<span class="at">pdata           =</span> <span class="cn">NULL</span>,</span>
<span id="cb335-2"><a href="tumor-ecosystem-analysis.html#cb335-2" tabindex="-1"></a>                             <span class="at">eset            =</span> eset,</span>
<span id="cb335-3"><a href="tumor-ecosystem-analysis.html#cb335-3" tabindex="-1"></a>                             <span class="at">signature       =</span> signature_collection,</span>
<span id="cb335-4"><a href="tumor-ecosystem-analysis.html#cb335-4" tabindex="-1"></a>                             <span class="at">method          =</span> <span class="st">&quot;pca&quot;</span>,</span>
<span id="cb335-5"><a href="tumor-ecosystem-analysis.html#cb335-5" tabindex="-1"></a>                             <span class="at">mini_gene_count =</span> <span class="dv">2</span>)</span>
<span id="cb335-6"><a href="tumor-ecosystem-analysis.html#cb335-6" tabindex="-1"></a>sig_tme <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">column_to_rownames</span>(sig_tme, <span class="at">var =</span> <span class="st">&quot;ID&quot;</span>))</span>
<span id="cb335-7"><a href="tumor-ecosystem-analysis.html#cb335-7" tabindex="-1"></a>sig_tme[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>##                   GSM1523727 GSM1523728 GSM1523729
## CD_8_T_effector   -2.5513794  0.7789141 -2.1770675
## DDR               -0.8747614  0.7425162 -1.3272054
## APM                1.1098368  2.1988688 -0.9516419
## Immune_Checkpoint -2.3701787  0.9455120 -1.4844104
## CellCycle_Reg      0.1063358  0.7583302 -0.3649795</code></pre>
<p>Finding signatures or cell types associated with TMEcluster</p>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb337-1"><a href="tumor-ecosystem-analysis.html#cb337-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">find_markers_in_bulk</span>(<span class="at">pdata =</span> tme, <span class="at">eset =</span> sig_tme, <span class="at">group =</span> <span class="st">&quot;TMEcluster&quot;</span>, <span class="at">nfeatures =</span> <span class="dv">1000</span>, <span class="at">top_n =</span> <span class="dv">20</span>, <span class="at">min.pct =</span> <span class="fl">0.10</span>)</span>
<span id="cb337-2"><a href="tumor-ecosystem-analysis.html#cb337-2" tabindex="-1"></a></span>
<span id="cb337-3"><a href="tumor-ecosystem-analysis.html#cb337-3" tabindex="-1"></a>top15 <span class="ot">&lt;-</span>  res<span class="sc">$</span>top_markers <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span> <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span>  dplyr<span class="sc">::</span><span class="fu">top_n</span>(<span class="dv">15</span>, avg_log2FC)</span>
<span id="cb337-4"><a href="tumor-ecosystem-analysis.html#cb337-4" tabindex="-1"></a></span>
<span id="cb337-5"><a href="tumor-ecosystem-analysis.html#cb337-5" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">DoHeatmap</span>(res<span class="sc">$</span>sce, top15<span class="sc">$</span>gene, <span class="at">group.colors =</span> cols)<span class="sc">+</span></span>
<span id="cb337-6"><a href="tumor-ecosystem-analysis.html#cb337-6" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">colours =</span> <span class="fu">rev</span>(<span class="fu">colorRampPalette</span>(RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">11</span>,<span class="st">&quot;RdBu&quot;</span>))(<span class="dv">256</span>)))</span></code></pre></div>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb338-1"><a href="tumor-ecosystem-analysis.html#cb338-1" tabindex="-1"></a>top15<span class="sc">$</span>gene  <span class="ot">&lt;-</span> <span class="fu">gsub</span>(top15<span class="sc">$</span>gene, <span class="at">pattern =</span> <span class="st">&quot;-&quot;</span>, <span class="at">replacement =</span> <span class="st">&quot;</span><span class="sc">\\</span><span class="st">_&quot;</span>)</span>
<span id="cb338-2"><a href="tumor-ecosystem-analysis.html#cb338-2" tabindex="-1"></a>input <span class="ot">&lt;-</span> <span class="fu">combine_pd_eset</span>(<span class="at">eset =</span> sig_tme, <span class="at">pdata =</span> tme, <span class="at">feas =</span> top15<span class="sc">$</span>gene, <span class="at">scale =</span> T)</span>
<span id="cb338-3"><a href="tumor-ecosystem-analysis.html#cb338-3" tabindex="-1"></a></span>
<span id="cb338-4"><a href="tumor-ecosystem-analysis.html#cb338-4" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">sig_box</span>(input, <span class="at">variable =</span> <span class="st">&quot;TMEcluster&quot;</span>, <span class="at">signature =</span> <span class="st">&quot;CD_8_T_effector&quot;</span>, <span class="at">jitter =</span> <span class="cn">TRUE</span>,</span>
<span id="cb338-5"><a href="tumor-ecosystem-analysis.html#cb338-5" tabindex="-1"></a>              <span class="at">cols =</span>  cols, <span class="at">show_pvalue =</span> <span class="cn">TRUE</span>, <span class="at">size_of_pvalue =</span> <span class="dv">4</span>, <span class="at">size_of_font =</span> <span class="dv">6</span>)</span></code></pre></div>
<pre><code>## # A tibble: 3 √ó 8
##   .y.       group1 group2        p    p.adj p.format p.signif method  
##   &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;     &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;   
## 1 signature IA     IE     3.53e-10 7.10e-10 3.5e-10  ****     Wilcoxon
## 2 signature IA     IS     8.49e-13 2.5 e-12 8.5e-13  ****     Wilcoxon
## 3 signature IE     IS     1.41e- 1 1.4 e- 1 0.14     ns       Wilcoxon</code></pre>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb340-1"><a href="tumor-ecosystem-analysis.html#cb340-1" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">sig_box</span>(input, <span class="at">variable =</span> <span class="st">&quot;TMEcluster&quot;</span>, <span class="at">signature =</span> <span class="st">&quot;Neutrophils_Bindea_et_al&quot;</span>,  </span>
<span id="cb340-2"><a href="tumor-ecosystem-analysis.html#cb340-2" tabindex="-1"></a>              <span class="at">jitter =</span> <span class="cn">TRUE</span>, <span class="at">cols =</span>  cols, <span class="at">show_pvalue =</span> <span class="cn">TRUE</span>, <span class="at">size_of_pvalue =</span> <span class="dv">4</span>, <span class="at">size_of_font =</span> <span class="dv">6</span>)</span></code></pre></div>
<pre><code>## # A tibble: 3 √ó 8
##   .y.       group1 group2         p   p.adj p.format p.signif method  
##   &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;      &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;   
## 1 signature IA     IE     0.00639   0.013   0.0064   **       Wilcoxon
## 2 signature IA     IS     0.0584    0.058   0.0584   ns       Wilcoxon
## 3 signature IE     IS     0.0000929 0.00028 9.3e-05  ****     Wilcoxon</code></pre>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb342-1"><a href="tumor-ecosystem-analysis.html#cb342-1" tabindex="-1"></a>p <span class="ot">&lt;-</span> (p1<span class="sc">|</span>p2<span class="sc">/</span>p3) <span class="sc">+</span> <span class="fu">plot_layout</span>(<span class="at">widths =</span> <span class="fu">c</span>(<span class="fl">2.3</span>,<span class="dv">1</span>))</span>
<span id="cb342-2"><a href="tumor-ecosystem-analysis.html#cb342-2" tabindex="-1"></a>p <span class="sc">+</span> <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">&#39;A&#39;</span>)</span></code></pre></div>
<p><img src="tme-deg-gsea_files/figure-html/unnamed-chunk-19-1.png" width="1152" /></p>
<div class="sourceCode" id="cb343"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb343-1"><a href="tumor-ecosystem-analysis.html#cb343-1" tabindex="-1"></a><span class="fu">library</span>(survminer)</span>
<span id="cb343-2"><a href="tumor-ecosystem-analysis.html#cb343-2" tabindex="-1"></a><span class="fu">data</span>(pdata_acrg, <span class="at">package =</span> <span class="st">&quot;IOBR&quot;</span>)</span>
<span id="cb343-3"><a href="tumor-ecosystem-analysis.html#cb343-3" tabindex="-1"></a>input <span class="ot">&lt;-</span> <span class="fu">merge</span>(pdata_acrg, input, <span class="at">by =</span> <span class="st">&quot;ID&quot;</span>)</span>
<span id="cb343-4"><a href="tumor-ecosystem-analysis.html#cb343-4" tabindex="-1"></a>p1<span class="ot">&lt;-</span><span class="fu">surv_group</span>(<span class="at">input_pdata       =</span> input,</span>
<span id="cb343-5"><a href="tumor-ecosystem-analysis.html#cb343-5" tabindex="-1"></a>               <span class="at">target_group      =</span> <span class="st">&quot;TMEcluster&quot;</span>,</span>
<span id="cb343-6"><a href="tumor-ecosystem-analysis.html#cb343-6" tabindex="-1"></a>               <span class="at">ID                =</span> <span class="st">&quot;ID&quot;</span>,</span>
<span id="cb343-7"><a href="tumor-ecosystem-analysis.html#cb343-7" tabindex="-1"></a>               <span class="at">reference_group   =</span> <span class="st">&quot;High&quot;</span>,</span>
<span id="cb343-8"><a href="tumor-ecosystem-analysis.html#cb343-8" tabindex="-1"></a>               <span class="at">project           =</span> <span class="st">&quot;ACRG&quot;</span>,</span>
<span id="cb343-9"><a href="tumor-ecosystem-analysis.html#cb343-9" tabindex="-1"></a>               <span class="at">cols              =</span> cols, </span>
<span id="cb343-10"><a href="tumor-ecosystem-analysis.html#cb343-10" tabindex="-1"></a>               <span class="at">time              =</span> <span class="st">&quot;OS_time&quot;</span>,</span>
<span id="cb343-11"><a href="tumor-ecosystem-analysis.html#cb343-11" tabindex="-1"></a>               <span class="at">status            =</span> <span class="st">&quot;OS_status&quot;</span>,</span>
<span id="cb343-12"><a href="tumor-ecosystem-analysis.html#cb343-12" tabindex="-1"></a>               <span class="at">time_type         =</span> <span class="st">&quot;month&quot;</span>,</span>
<span id="cb343-13"><a href="tumor-ecosystem-analysis.html#cb343-13" tabindex="-1"></a>               <span class="at">save_path         =</span> <span class="st">&quot;result&quot;</span>)</span></code></pre></div>
<pre><code>## &gt;&gt;&gt; Dataset&#39;s survival follow up time is range between 1 to 105.7 months</code></pre>
<pre><code>##  IA  IE  IS 
## 107  96  97</code></pre>
<pre><code>## 1079697</code></pre>
<pre><code>##   Maximum of follow up time is 105.7 months; and will be divided into 6 sections;</code></pre>
<pre><code>## Warning in geom_segment(aes(x = 0, y = max(y2), xend = max(x1), yend = max(y2)), : All aesthetics have length 1, but the data has 2 rows.
## ‚Ñπ Please consider using `annotate()` or provide this layer with data containing
##   a single row.</code></pre>
<pre><code>## Registered S3 methods overwritten by &#39;ggpp&#39;:
##   method                  from   
##   heightDetails.titleGrob ggplot2
##   widthDetails.titleGrob  ggplot2</code></pre>
<pre><code>## Warning in geom_segment(aes(x = 0, y = max(y2), xend = max(x1), yend = max(y2)), : All aesthetics have length 1, but the data has 2 rows.
## ‚Ñπ Please consider using `annotate()` or provide this layer with data containing
##   a single row.
## All aesthetics have length 1, but the data has 2 rows.
## ‚Ñπ Please consider using `annotate()` or provide this layer with data containing
##   a single row.
## All aesthetics have length 1, but the data has 2 rows.
## ‚Ñπ Please consider using `annotate()` or provide this layer with data containing
##   a single row.</code></pre>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="tumor-ecosystem-analysis.html#cb351-1" tabindex="-1"></a>p1</span></code></pre></div>
<p><img src="tme-deg-gsea_files/figure-html/unnamed-chunk-20-1.png" width="480" /></p>
<div class="sourceCode" id="cb352"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb352-1"><a href="tumor-ecosystem-analysis.html#cb352-1" tabindex="-1"></a>p1<span class="ot">&lt;-</span> <span class="fu">percent_bar_plot</span>(input, <span class="at">x =</span> <span class="st">&quot;TMEcluster&quot;</span> , <span class="at">y =</span> <span class="st">&quot;Subtype&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;jama&quot;</span>)</span></code></pre></div>
<pre><code>## # A tibble: 12 √ó 5
## # Groups:   TMEcluster [3]
##    TMEcluster Subtype    Freq  Prop count
##    &lt;chr&gt;      &lt;fct&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
##  1 IA         EMT           7  0.07   107
##  2 IA         MSI          49  0.46   107
##  3 IA         MSS/TP53-    27  0.25   107
##  4 IA         MSS/TP53+    24  0.22   107
##  5 IE         EMT          24  0.25    96
##  6 IE         MSI           3  0.03    96
##  7 IE         MSS/TP53-    40  0.42    96
##  8 IE         MSS/TP53+    29  0.3     96
##  9 IS         EMT          15  0.15    97
## 10 IS         MSI          16  0.16    97
## 11 IS         MSS/TP53-    40  0.41    97
## 12 IS         MSS/TP53+    26  0.27    97</code></pre>
<pre><code>## [1] &quot;&#39;#374E55FF&#39;, &#39;#DF8F44FF&#39;, &#39;#00A1D5FF&#39;, &#39;#B24745FF&#39;, &#39;#79AF97FF&#39;, &#39;#6A6599FF&#39;, &#39;#80796BFF&#39;&quot;</code></pre>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb355-1"><a href="tumor-ecosystem-analysis.html#cb355-1" tabindex="-1"></a>p2<span class="ot">&lt;-</span> <span class="fu">percent_bar_plot</span>(input, <span class="at">x =</span> <span class="st">&quot;TMEcluster&quot;</span> , <span class="at">y =</span> <span class="st">&quot;Lauren&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;jama&quot;</span>)</span></code></pre></div>
<pre><code>## # A tibble: 9 √ó 5
## # Groups:   TMEcluster [3]
##   TMEcluster Lauren      Freq  Prop count
##   &lt;chr&gt;      &lt;fct&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
## 1 IA         Diffuse       34  0.32   107
## 2 IA         Intestinal    60  0.56   107
## 3 IA         Mixed         13  0.12   107
## 4 IE         Diffuse       60  0.62    96
## 5 IE         Intestinal    32  0.33    96
## 6 IE         Mixed          4  0.04    96
## 7 IS         Diffuse       41  0.42    97
## 8 IS         Intestinal    54  0.56    97
## 9 IS         Mixed          2  0.02    97</code></pre>
<pre><code>## [1] &quot;&#39;#374E55FF&#39;, &#39;#DF8F44FF&#39;, &#39;#00A1D5FF&#39;, &#39;#B24745FF&#39;, &#39;#79AF97FF&#39;, &#39;#6A6599FF&#39;, &#39;#80796BFF&#39;&quot;</code></pre>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb358-1"><a href="tumor-ecosystem-analysis.html#cb358-1" tabindex="-1"></a>p3<span class="ot">&lt;-</span> <span class="fu">percent_bar_plot</span>(input, <span class="at">x =</span> <span class="st">&quot;TMEcluster&quot;</span> , <span class="at">y =</span> <span class="st">&quot;TMEscore_binary&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;jama&quot;</span>)</span></code></pre></div>
<pre><code>## # A tibble: 7 √ó 5
## # Groups:   TMEcluster [3]
##   TMEcluster TMEscore_binary  Freq  Prop count
##   &lt;chr&gt;      &lt;fct&gt;           &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
## 1 IA         High               60  0.56   107
## 2 IA         Low                47  0.44   107
## 3 IE         High                5  0.05    96
## 4 IE         Low                91  0.95    96
## 5 IS         High                6  0.06    97
## 6 IS         Low                90  0.93    97
## 7 IS         &lt;NA&gt;                1  0.01    97</code></pre>
<pre><code>## [1] &quot;&#39;#374E55FF&#39;, &#39;#DF8F44FF&#39;, &#39;#00A1D5FF&#39;, &#39;#B24745FF&#39;, &#39;#79AF97FF&#39;, &#39;#6A6599FF&#39;, &#39;#80796BFF&#39;&quot;</code></pre>
<div class="sourceCode" id="cb361"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb361-1"><a href="tumor-ecosystem-analysis.html#cb361-1" tabindex="-1"></a>p1<span class="sc">|</span>p2<span class="sc">|</span>p3</span></code></pre></div>
<p><img src="tme-deg-gsea_files/figure-html/unnamed-chunk-22-1.png" width="1152" /></p>
</div>
<div id="references-4" class="section level2 hasAnchor" number="8.9">
<h2><span class="header-section-number">8.9</span> References<a href="tumor-ecosystem-analysis.html#references-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Cristescu, R., Lee, J., Nebozhyn, M. et al.¬†Molecular analysis of gastric cancer identifies subtypes associated with distinct clinical outcomes. Nat Med 21, 449‚Äì456 (2015). <a href="https://doi.org/10.1038/nm.3850" class="uri">https://doi.org/10.1038/nm.3850</a></p>
<p><a href="https://cibersort.stanford.edu/">CIBERSORT</a>; Newman, A. M., Liu, C. L., Green, M. R., Gentles, A. J., Feng, W., Xu, Y., ‚Ä¶ Alizadeh, A. A. (2015). Robust enumeration of cell subsets from tissue expression profiles. Nature Methods, 12(5), 453‚Äì457. <a href="https://doi.org/10.1038/nmeth.3337" class="uri">https://doi.org/10.1038/nmeth.3337</a>;</p>
<p>Seurat: Hao and Hao et al.¬†Integrated analysis of multimodal single-cell data. Cell (2021)</p>
<p>Zeng D, Yu Y, Qiu W, Mao Q, ‚Ä¶, Zhang K, Liao W; Tumor microenvironment immunotyping heterogeneity reveals distinct molecular mechanisms to clinical immunotherapy applications in gastric cancer. (2024) Under Review.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="tme-interaction-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="tme-and-genomic-interaction.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/DongqiangZeng0808/tme-deg-gsea.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
